// @notus-pro/react/Accordion.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Accordion;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var AccordionItem = function AccordionItem(_ref) {
  var title = _ref.title,
      color = _ref.color,
      description = _ref.description,
      defaultOpened = _ref.defaultOpened,
      setOpen = _ref.setOpen;

  var _React$useState = _react.default.useState(defaultOpened),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      collapseOpen = _React$useState2[0],
      setCollapseOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(defaultOpened),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      rotate = _React$useState4[0],
      setRotate = _React$useState4[1];

  var _React$useState5 = _react.default.useState(undefined),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      collapseStyle = _React$useState6[0],
      setCollapseStyle = _React$useState6[1];

  var _React$useState7 = _react.default.useState(false),
      _React$useState8 = _slicedToArray(_React$useState7, 2),
      animation = _React$useState8[0],
      setAnimation = _React$useState8[1];

  var collapseRef = _react.default.useRef(null);

  var openAnimation = function openAnimation() {
    setOpen();

    if (!collapseOpen && collapseStyle === undefined) {
      setCollapseStyle(0);
    }

    setCollapseOpen(true);
    setTimeout(function () {
      setCollapseStyle(collapseRef.current.scrollHeight);
    }, 10);
    setTimeout(function () {
      setAnimation(false);
    }, 310);
  };

  var closeAnimation = function closeAnimation() {
    var timeOutTime = 0;

    if (collapseOpen && collapseStyle === undefined) {
      setCollapseStyle(collapseRef.current.scrollHeight);
      timeOutTime = 10;
    }

    setTimeout(function () {
      setCollapseStyle(0);
    }, timeOutTime);
    setTimeout(function () {
      setAnimation(false);
      setCollapseOpen(false);
    }, 310);
  };

  var startAnitmation = function startAnitmation(e) {
    e.preventDefault();

    if (!animation) {
      setAnimation(true);
      setRotate(!rotate);

      if (collapseOpen) {
        closeAnimation();
      } else {
        openAnimation();
      }
    }
  };

  _react.default.useEffect(function () {
    if (!defaultOpened) {
      setCollapseStyle(collapseRef.current.scrollHeight);
      setRotate(false);
      setTimeout(function () {
        setCollapseStyle(0);
      }, 10);
      setTimeout(function () {
        setAnimation(false);
        setCollapseOpen(false);
      }, 310);
    }
  }, [defaultOpened]);

  var colors = {
    blueGray: "text-blueGray-700 hover:text-blueGray-900",
    red: "text-red-500 hover:text-red-700",
    orange: "text-orange-500 hover:text-orange-700",
    amber: "text-amber-500 hover:text-amber-700",
    emerald: "text-emerald-500 hover:text-emerald-700",
    teal: "text-teal-500 hover:text-teal-700",
    lightBlue: "text-lightBlue-500 hover:text-lightBlue-700",
    indigo: "text-indigo-500 hover:text-indigo-700",
    purple: "text-purple-500 hover:text-purple-700",
    pink: "text-pink-500 hover:text-pink-700"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-transparent first:rounded-t px-4 py-3"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#openCollapse",
    onClick: startAnitmation
  }, /*#__PURE__*/_react.default.createElement("h5", {
    className: colors[color] + " mb-0 font-semibold duration-300 transition-all ease-in-out"
  }, title, /*#__PURE__*/_react.default.createElement("i", {
    className: (0, _classnames.default)("text-sm mr-2 float-right fas fa-chevron-down duration-300 transition-all ease-in-out transform", {
      "rotate-180": rotate
    })
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)("duration-300 transition-all ease-in-out", {
      block: collapseOpen,
      hidden: !collapseOpen
    }),
    style: {
      height: collapseStyle
    },
    ref: collapseRef
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-blueGray-500 px-4 py-5 flex-auto leading-relaxed"
  }, typeof description === "string" && /*#__PURE__*/_react.default.createElement("p", {
    className: "last:mb-2"
  }, description), Array.isArray(description) && description.map(function (descriptionProp, descriptionKey) {
    return /*#__PURE__*/_react.default.createElement("p", {
      key: descriptionKey,
      className: "last:mb-2"
    }, descriptionProp);
  }))));
};

AccordionItem.defaultProps = {
  defaultOpened: false,
  setOpen: function setOpen() {}
};

function Accordion(_ref2) {
  var items = _ref2.items,
      color = _ref2.color,
      defaultOpened = _ref2.defaultOpened;

  var _React$useState9 = _react.default.useState(defaultOpened),
      _React$useState10 = _slicedToArray(_React$useState9, 2),
      open = _React$useState10[0],
      setOpen = _React$useState10[1];

  var callBackAction = function callBackAction(key) {
    if (!Array.isArray(defaultOpened)) {
      setOpen(key);
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", null, items.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "overflow-hidden relative flex flex-col min-w-0 break-words bg-white w-full mb-5 border-b border-blueGray-200"
    }, /*#__PURE__*/_react.default.createElement(AccordionItem, {
      title: prop.title,
      description: prop.description,
      color: color,
      setOpen: function setOpen() {
        return callBackAction(key);
      },
      defaultOpened: key === open || Array.isArray(open) && open.includes(key)
    }));
  })));
}

Accordion.defaultProps = {
  items: [],
  defaultOpened: -1,
  multiple: false
};
Accordion.propTypes = {
  // NOTE: if you pass an array for the defaultOpened prop
  // // // then the user will be able to open multiple collapses
  // // // For example, if you want to have only the first
  // // // collapse opened, but the user can open multiple
  // // // then you can pass defaultOpened={[0]}
  // // // otherwise, you can pass defaultOpened={0}
  defaultOpened: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.arrayOf(_propTypes.default.number)]),
  items: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    description: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.arrayOf(_propTypes.default.string)])
  })),
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
