// @notus-pro/react/Admin2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Admin2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Sidebar = _interopRequireDefault(require("./Sidebar.js"));

var _NavbarSearchUser = _interopRequireDefault(require("./NavbarSearchUser.js"));

var _HeaderUser = _interopRequireDefault(require("./HeaderUser.js"));

var _FooterAdmin = _interopRequireDefault(require("./FooterAdmin.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Admin2(_ref) {
  var sidebar = _ref.sidebar,
      navbar = _ref.navbar,
      header = _ref.header,
      footer = _ref.footer,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Sidebar.default, sidebar), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative md:ml-64 bg-blueGray-200"
  }, /*#__PURE__*/_react.default.createElement(_NavbarSearchUser.default, navbar), /*#__PURE__*/_react.default.createElement(_HeaderUser.default, header), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 md:px-10 mx-auto w-full -mt-24"
  }, children ? children : /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full h-500-px bg-white text-center flex flex-col items-center justify-center content-center border border-solid border-blueGray-200 rounded p-40"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-6xl font-normal leading-normal mt-0 mb-2"
  }, "Here will be your content, instead of this box!"), /*#__PURE__*/_react.default.createElement("p", null, "You can add one of the width utility classes to make it a little smaller.")), /*#__PURE__*/_react.default.createElement(_FooterAdmin.default, footer))));
}

Admin2.defaultProps = {
  sidebar: {},
  navbar: {},
  header: {},
  footer: {}
};
Admin2.propTypes = {
  // props to pass to the Sidebar component
  sidebar: _propTypes.default.object,
  // props to pass to the NavbarSearchUser component
  navbar: _propTypes.default.object,
  // props to pass to the HeaderUser component
  header: _propTypes.default.object,
  // props to pass to the FooterAdmin component
  footer: _propTypes.default.object
};
