// @notus-pro/react/Auth1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Auth1;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _NavbarLinks = _interopRequireDefault(require("./NavbarLinks.js"));

var _FooterAuth = _interopRequireDefault(require("./FooterAuth.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Auth1(_ref) {
  var children = _ref.children,
      footer = _ref.footer,
      navbar = _ref.navbar,
      image = _ref.image;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, navbar ? /*#__PURE__*/_react.default.createElement(_NavbarLinks.default, navbar) : null, /*#__PURE__*/_react.default.createElement("main", null, /*#__PURE__*/_react.default.createElement("section", {
    className: "relative w-full h-full py-40 min-h-screen"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute top-0 w-full h-full bg-blueGray-900 bg-no-repeat bg-full",
    style: {
      backgroundImage: "url(" + image + ")"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 h-full"
  }, children ? children : /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full h-600-px bg-white text-center flex flex-col items-center justify-center content-center border border-solid border-blueGray-200 rounded p-40"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-6xl font-normal leading-normal mt-0 mb-2"
  }, "Here will be your content, instead of this box!"), /*#__PURE__*/_react.default.createElement("p", null, "You can add one of the width utility classes to make it a little smaller."))), footer ? /*#__PURE__*/_react.default.createElement(_FooterAuth.default, footer) : null)));
}

Auth1.defaultProps = {
  children: undefined
};
Auth1.propTypes = {
  children: _propTypes.default.node,
  image: _propTypes.default.string,
  // props that you can pass to the NavbarLinks component
  // NOTE: if you do not pass this object,
  // // // the navbar will not be rendered
  navbar: _propTypes.default.object,
  // props that you can pass to the FooterAuth component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  footer: _propTypes.default.object
};
