// @notus-pro/react/Auth2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Auth2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _NavbarLinks = _interopRequireDefault(require("./NavbarLinks.js"));

var _HeaderImageTitleSmallCenter = _interopRequireDefault(require("./HeaderImageTitleSmallCenter.js"));

var _PreFooterLarge = _interopRequireDefault(require("./PreFooterLarge.js"));

var _FooterSmall = _interopRequireDefault(require("./FooterSmall.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Auth2(_ref) {
  var children = _ref.children,
      footer = _ref.footer,
      navbar = _ref.navbar,
      header = _ref.header,
      prefooter = _ref.prefooter;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, navbar ? /*#__PURE__*/_react.default.createElement(_NavbarLinks.default, navbar) : null, header ? /*#__PURE__*/_react.default.createElement(_HeaderImageTitleSmallCenter.default, header) : null, /*#__PURE__*/_react.default.createElement("section", {
    className: "relative bg-blueGray-100 py-12 z-2"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, children ? children : /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full h-600-px bg-white text-center flex flex-col items-center justify-center content-center border border-solid border-blueGray-200 rounded p-40"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-6xl font-normal leading-normal mt-0 mb-2"
  }, "Here will be your content, instead of this box!"), /*#__PURE__*/_react.default.createElement("p", null, "You can add one of the width utility classes to make it a little smaller, and also, you can add one of the negative margin utility classes to make it render where the skew starts")))), prefooter ? /*#__PURE__*/_react.default.createElement(_PreFooterLarge.default, prefooter) : null, footer ? /*#__PURE__*/_react.default.createElement(_FooterSmall.default, footer) : null);
}

Auth2.defaultProps = {
  children: undefined
};
Auth2.propTypes = {
  children: _propTypes.default.node,
  // props that you can pass to the NavbarLinks component
  // NOTE: if you do not pass this object,
  // // // the navbar will not be rendered
  navbar: _propTypes.default.object,
  // props that you can pass to the HeaderImageTitleSmallCenter component
  // NOTE: if you do not pass this object,
  // // // the navbar will not be rendered
  header: _propTypes.default.object,
  // props that you can pass to the PreFooterLarge component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  prefooter: _propTypes.default.object,
  // props that you can pass to the FooterSmall component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  footer: _propTypes.default.object
};
