// @notus-pro/react/Auth3.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Auth3;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _NavbarLinks = _interopRequireDefault(require("./NavbarLinks.js"));

var _PreFooterLarge = _interopRequireDefault(require("./PreFooterLarge.js"));

var _FooterSmall = _interopRequireDefault(require("./FooterSmall.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Auth3(_ref) {
  var children = _ref.children,
      footer = _ref.footer,
      navbar = _ref.navbar,
      prefooter = _ref.prefooter,
      color = _ref.color;
  var bgColor = {
    "purple-indigo": "from-purple-900 to-indigo-900 bg-indigo-900 bg-indigo-900",
    "orange-red": "from-orange-900 to-red-900 bg-red-900 bg-red-900",
    "amber-orange": "from-amber-900 to-orange-900 bg-orange-900 bg-orange-900",
    "emerald-teal": "from-emerald-900 to-teal-900 bg-teal-900 bg-teal-900",
    "lightBlue-indigo": "from-lightBlue-900 to-indigo-900 bg-indigo-900 bg-indigo-900",
    "pink-purple": "from-pink-900 to-purple-900 bg-purple-900 bg-purple-900"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, navbar ? /*#__PURE__*/_react.default.createElement(_NavbarLinks.default, navbar) : null, /*#__PURE__*/_react.default.createElement("main", null, /*#__PURE__*/_react.default.createElement("section", {
    className: "relative w-full h-full py-40 min-h-screen"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute top-0 w-full h-full bg-no-repeat bg-full bg-gradient-to-r  " + bgColor[color]
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-10 w-32 h-32 rounded-full -left-1/25 bottom-auto"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-10 w-32 h-32 rounded-full right-1/25 top-10"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-30 w-32 h-32 rounded-full top-280-px right-1/20"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-15 w-32 h-32 rounded-full top-320-px right-7/100"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-5 w-32 h-32 rounded-full top-19/50 left-1/100 right-auto"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-15 w-48 h-48 rounded-full top-11/25 left-1/10 right-auto"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-4 w-32 h-32 rounded-full bottom-1/2 right-9/25"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "absolute bg-white opacity-20 w-32 h-32 rounded-full botom-70-px right-1/50"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 h-full"
  }, children ? children : /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full h-600-px bg-white text-center flex flex-col items-center justify-center content-center border border-solid border-blueGray-200 rounded p-40"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-6xl font-normal leading-normal mt-0 mb-2"
  }, "Here will be your content, instead of this box!"), /*#__PURE__*/_react.default.createElement("p", null, "You can add one of the width utility classes to make it a little smaller."))))), prefooter ? /*#__PURE__*/_react.default.createElement(_PreFooterLarge.default, prefooter) : null, footer ? /*#__PURE__*/_react.default.createElement(_FooterSmall.default, footer) : null);
}

Auth3.defaultProps = {
  children: undefined,
  color: "purple-indigo"
};
Auth3.propTypes = {
  children: _propTypes.default.node,
  color: _propTypes.default.oneOf(["purple-indigo", "orange-red", "amber-orange", "emerald-teal", "lightBlue-indigo", "pink-purple"]),
  // props that you can pass to the NavbarLinks component
  // NOTE: if you do not pass this object,
  // // // the navbar will not be rendered
  navbar: _propTypes.default.object,
  // props that you can pass to the PreFooterLarge component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  prefooter: _propTypes.default.object,
  // props that you can pass to the FooterSmall component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  footer: _propTypes.default.object
};
