// @notus-pro/react/Auth4.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Auth4;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _NavbarLinks = _interopRequireDefault(require("./NavbarLinks.js"));

var _PreFooterLarge = _interopRequireDefault(require("./PreFooterLarge.js"));

var _FooterSmall = _interopRequireDefault(require("./FooterSmall.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Auth4(_ref) {
  var children = _ref.children,
      footer = _ref.footer,
      navbar = _ref.navbar,
      prefooter = _ref.prefooter,
      image = _ref.image;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, navbar ? /*#__PURE__*/_react.default.createElement(_NavbarLinks.default, navbar) : null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundImage: "url('" + image + "')"
    },
    className: "bg-blueGray-800 relative py-24 bg-cover bg-50"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-12 container mx-auto px-4"
  }, children ? children : /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full h-600-px bg-white text-center flex flex-col items-center justify-center content-center border border-solid border-blueGray-200 rounded p-40"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-6xl font-normal leading-normal mt-0 mb-2"
  }, "Here will be your content, instead of this box!"), /*#__PURE__*/_react.default.createElement("p", null, "You can add one of the width utility classes to make it a little smaller.")))), prefooter ? /*#__PURE__*/_react.default.createElement(_PreFooterLarge.default, prefooter) : null, footer ? /*#__PURE__*/_react.default.createElement(_FooterSmall.default, footer) : null);
}

Auth4.defaultProps = {
  children: undefined
};
Auth4.propTypes = {
  children: _propTypes.default.node,
  image: _propTypes.default.string,
  // props that you can pass to the NavbarLinks component
  // NOTE: if you do not pass this object,
  // // // the navbar will not be rendered
  navbar: _propTypes.default.object,
  // props that you can pass to the PreFooterLarge component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  prefooter: _propTypes.default.object,
  // props that you can pass to the FooterSmall component
  // NOTE: if you do not pass this object,
  // // // the footer will not be rendered
  footer: _propTypes.default.object
};
