// @notus-pro/react/Blogs4.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Blogs4;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _MediaPlayerImageOnly = _interopRequireDefault(require("./MediaPlayerImageOnly.js"));

var _ImagePost = _interopRequireDefault(require("./ImagePost.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Blogs4(_ref) {
  var items = _ref.items;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, items.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("section", {
      key: key,
      className: "pt-12 relative"
    }, prop.imagepost ? /*#__PURE__*/_react.default.createElement("div", {
      className: "container mx-auto px-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex flex-wrap -mx-4 justify-center"
    }, /*#__PURE__*/_react.default.createElement(_ImagePost.default, prop.imagepost))) : null, prop.mediaplayer ? /*#__PURE__*/_react.default.createElement(_MediaPlayerImageOnly.default, prop.mediaplayer) : null, prop.textSection ? /*#__PURE__*/_react.default.createElement("div", {
      className: "container mx-auto px-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex flex-wrap -mx-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "mx-auto relative w-full md:w-8/12"
    }, prop.textSection.map(function (textProp, textKey) {
      if (textProp.title) {
        return /*#__PURE__*/_react.default.createElement("h3", {
          key: textKey,
          className: "text-3xl font-bold leading-normal mt-0 mb-2"
        }, textProp.title);
      } else if (textProp.subtitle) {
        return /*#__PURE__*/_react.default.createElement("h6", {
          key: textKey,
          className: "text-lg mt-2 mb-0 text-blueGray-400"
        }, textProp.subtitle);
      } else if (textProp.quote) {
        return /*#__PURE__*/_react.default.createElement("p", {
          key: textKey,
          className: "text-xl text-blueGray-500 mx-0 my-12 block pl-4 border-l border-blueGray-200 border-l-2"
        }, "\"", textProp.quote, "\"", /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("small", {
          className: "mt-2 font-semibold text-blueGray-700"
        }, textProp.author));
      } else {
        return /*#__PURE__*/_react.default.createElement("p", {
          key: textKey,
          className: "mb-4 text-lg text-blueGray-500"
        }, textProp.description);
      }
    })))) : null);
  }));
}

Blogs4.defaultProps = {
  items: []
};
Blogs4.propTypes = {
  items: _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.shape({
    // an object representing the props
    // that can be passed to our ImagePost component
    imagepost: _propTypes.default.object
  }), _propTypes.default.shape({
    // an object representing the props
    // that can be passed to our MediaPlayerImageOnly component
    mediaplayer: _propTypes.default.object
  }), _propTypes.default.shape({
    textSection: _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.shape({
      title: _propTypes.default.string
    }), _propTypes.default.shape({
      subtitle: _propTypes.default.string
    }), _propTypes.default.shape({
      description: _propTypes.default.string
    }), _propTypes.default.shape({
      quote: _propTypes.default.string,
      author: _propTypes.default.string
    })]))
  })]))
};
