// @notus-pro/react/Blogs9.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Blogs9;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _Heading3Badge = _interopRequireDefault(require("./Heading3Badge.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Blogs9(_ref) {
  var heading3badge = _ref.heading3badge,
      user = _ref.user,
      leftButtons = _ref.leftButtons,
      image = _ref.image,
      description = _ref.description,
      links = _ref.links,
      bottomButtons = _ref.bottomButtons;
  var newDescription = [];

  if (!Array.isArray(description)) {
    newDescription = [description];
  } else {
    newDescription = description;
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading3badge && /*#__PURE__*/_react.default.createElement(_Heading3Badge.default, heading3badge), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto px-4 relative w-full lg:w-10/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center justify-between my-4 flex-col md:flex-row"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center mb-4 md:mb-0"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: user.image,
    alt: "...",
    className: "rounded-full mr-4 max-w-40-px"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between items-center"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-lg font-bold leading-tight"
  }, user.name), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-500 text-sm"
  }, user.date)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative inline-block"
  }, leftButtons.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      key: key
    }, prop));
  }))), /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "w-full rounded-lg shadow-lg"
  }), newDescription.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("p", {
      key: key,
      className: "text-blueGray-500 pt-6 leading-relaxed"
    }, prop);
  }), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-lg font-normal mt-2 mb-4 text-lightBlue-500"
  }, links.map(function (prop, key) {
    if (prop && prop.to) {
      return /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({
        key: key
      }, prop, {
        className: "mr-2"
      }));
    }

    return /*#__PURE__*/_react.default.createElement("a", _extends({
      key: key
    }, prop, {
      className: "mr-2"
    }));
  }))), bottomButtons.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      key: key
    }, prop));
  }))))));
}

Blogs9.defaultProps = {
  user: {},
  description: [],
  leftButtons: [],
  bottomButtons: [],
  links: [],
  heading3badge: null
};
Blogs9.propTypes = {
  // prop to pass to the Heading3Badge component
  heading3badge: _propTypes.default.object,
  user: _propTypes.default.shape({
    image: _propTypes.default.string,
    name: _propTypes.default.string,
    date: _propTypes.default.string
  }),
  leftButtons: _propTypes.default.arrayOf(_propTypes.default.object),
  image: _propTypes.default.string,
  description: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.arrayOf(_propTypes.default.string)]),
  // array of properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  links: _propTypes.default.arrayOf(_propTypes.default.object),
  // array of props to pass to the Button component
  bottomButtons: _propTypes.default.arrayOf(_propTypes.default.object)
};
