// @notus-pro/react/Blogs7.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Blogs7;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading2Badge = _interopRequireDefault(require("./Heading2Badge.js"));

var _CardBlogFullBg = _interopRequireDefault(require("./CardBlogFullBg.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Blogs7(_ref) {
  var heading2badge = _ref.heading2badge,
      cards = _ref.cards,
      button = _ref.button;
  var widths = {
    1: "",
    "1/3": "lg:w-1/3",
    "2/3": "lg:w-2/3",
    "1/4": "lg:w-1/4",
    "2/4": "lg:w-2/4",
    "3/4": "lg:w-3/4"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "my-12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12"
  }, heading2badge && /*#__PURE__*/_react.default.createElement(_Heading2Badge.default, heading2badge)), /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex flex-wrap -mx-4 mb-0"
  }, cards.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "px-4 mb-4 relative w-full " + widths[prop.width]
    }, /*#__PURE__*/_react.default.createElement(_CardBlogFullBg.default, prop.card));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button))));
}

Blogs7.defaultProps = {
  cards: [],
  heading2badge: null
};
Blogs7.propTypes = {
  // prop to pass to the Heading2Badge component
  heading2badge: _propTypes.default.object,
  // props that can be passed to the Button component
  button: _propTypes.default.object,
  cards: _propTypes.default.arrayOf(_propTypes.default.shape({
    width: _propTypes.default.oneOf(["1", "1/3", "2/3", "1/4", "2/4", "3/4"]),
    // props that can be passed to the CardBlogFullBg component
    card: _propTypes.default.object
  }))
};
