// @notus-pro/react/CardProductText.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProductText;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardProductText(_ref) {
  var image = _ref.image,
      icon = _ref.icon,
      color = _ref.color,
      text = _ref.text,
      description = _ref.description,
      title = _ref.title,
      link = _ref.link;
  var iconColor = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "shadow-lg rounded-lg overflow-hidden relative flex flex-col min-w-0 break-words bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-6"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "w-full"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-6 pb-6 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: iconColor[color] + " font-bold uppercase text-sm mb-2"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: icon + " mr-1 opacity-75"
  }), " ", text), /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-2xl font-semibold leading-tight mt-0 mb-2 my-4"
  }, link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, link, {
    className: ""
  }), title) : /*#__PURE__*/_react.default.createElement("a", _extends({}, link, {
    className: ""
  }), title)), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-500 leading-relaxed"
  }, description))));
}

CardProductText.defaultProps = {
  link: {},
  color: "blueGray"
};
CardProductText.propTypes = {
  image: _propTypes.default.string,
  icon: _propTypes.default.string,
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  text: _propTypes.default.string,
  description: _propTypes.default.string,
  title: _propTypes.default.string,
  // properties to pass to the link object that wraps the title
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object
};
