// @notus-pro/react/CardPricingList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardPricingList;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardPricingList(_ref) {
  var head = _ref.head,
      title = _ref.title,
      list = _ref.list,
      gradient = _ref.gradient,
      color = _ref.color,
      image = _ref.image,
      button = _ref.button;
  var colors = {
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: colors[color] + " pt-12 mb-6 text-center text-white shadow-lg rounded-lg relative flex flex-col min-w-0 break-words w-full mb-6 bg-contain bg-no-repeat bg-50-top",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-12 px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "font-bold uppercase mb-2"
  }, head), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-6xl font-bold leading-tight mt-0 mb-2"
  }, title), /*#__PURE__*/_react.default.createElement("ul", {
    className: "my-6 mx-4 list-none"
  }, list.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("li", {
      key: key,
      className: "py-3 border-b border-black border-opacity-10"
    }, /*#__PURE__*/_react.default.createElement("b", {
      className: "mr-2"
    }, prop.title), prop.description);
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-6"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button)))));
}

CardPricingList.defaultProps = {
  list: [],
  button: {}
};
CardPricingList.propTypes = {
  image: _propTypes.default.string,
  head: _propTypes.default.string,
  title: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object,
  list: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    description: _propTypes.default.string
  })),
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
