// @notus-pro/react/CardLoginRegister.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardLoginRegister;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function CardLoginRegister(_ref) {
  var leftCard = _ref.leftCard,
      rightCard = _ref.rightCard,
      detailsPanel = _ref.detailsPanel,
      color = _ref.color;

  var _React$useState = _react.default.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      animating = _React$useState2[0],
      setAntimating = _React$useState2[1];

  var _React$useState3 = _react.default.useState(""),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      leftCardTransform = _React$useState4[0],
      setLeftCardTransform = _React$useState4[1];

  var _React$useState5 = _react.default.useState("z-1 opacity-0"),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      rightCardTransform = _React$useState6[0],
      setRightCardTransform = _React$useState6[1];

  var _React$useState7 = _react.default.useState(""),
      _React$useState8 = _slicedToArray(_React$useState7, 2),
      panelTransform = _React$useState8[0],
      setPanelTransform = _React$useState8[1];

  var _React$useState9 = _react.default.useState(""),
      _React$useState10 = _slicedToArray(_React$useState9, 2),
      panelInnerTransform = _React$useState10[0],
      setPanelInnerTransform = _React$useState10[1];

  var _React$useState11 = _react.default.useState(""),
      _React$useState12 = _slicedToArray(_React$useState11, 2),
      panelLeftTransform = _React$useState12[0],
      setPanelLeftTransform = _React$useState12[1];

  var _React$useState13 = _react.default.useState("-translate-x-1/5"),
      _React$useState14 = _slicedToArray(_React$useState13, 2),
      panelRightTransform = _React$useState14[0],
      setPanelRightTransform = _React$useState14[1];

  var animation = function animation() {
    if (!animating) {
      setAntimating(true);
      panelTransform === "" ? setPanelTransform("-translate-x-full") : setPanelTransform("");
      panelInnerTransform === "" ? setPanelInnerTransform("translate-x-1/2") : setPanelInnerTransform("");
      panelRightTransform === "" ? setPanelRightTransform("-translate-x-1/5") : setPanelRightTransform("");
      panelLeftTransform === "" ? setPanelLeftTransform("translate-x-1/5") : setPanelLeftTransform("");
      leftCardTransform === "" ? setLeftCardTransform("translate-x-full") : setLeftCardTransform("");
      rightCardTransform === "z-1 opacity-0" ? setRightCardTransform("z-5 opcity-1 translate-x-full") : setRightCardTransform("z-1 opacity-0");
      setTimeout(function () {
        setAntimating(false);
      }, 710);
    }
  };

  var panelColorClass = {
    light: "bg-blueGray-200 text-blueGray-700",
    dark: "bg-blueGray-800 text-white",
    blueGray: "bg-blueGray-500 text-white",
    red: "bg-red-500 text-white",
    orange: "bg-orange-500 text-white",
    amber: "bg-amber-500 text-white",
    emerald: "bg-emerald-500 text-white",
    teal: "bg-teal-500 text-white",
    lightBlue: "bg-lightBlue-500 text-white",
    indigo: "bg-indigo-500 text-white",
    purple: "bg-purple-500 text-white",
    pink: "bg-pink-500 text-white"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-200 rounded-lg shadow-lg relative overflow-hidden h-full z-1 max-w-full min-h-480-px w-768-px container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "transition-all bg-blueGray-200 duration-700 ease-in-out h-full top-0 absolute w-1/2 left-0 z-2 transform " + leftCardTransform
  }, /*#__PURE__*/_react.default.createElement("form", {
    className: "flex items-center justify-center flex-col px-12 h-full text-center"
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: "text-2xp mb-3 font-bold"
  }, leftCard.title), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center"
  }, leftCard.socials.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      key: key
    }, prop.button), /*#__PURE__*/_react.default.createElement("i", {
      className: prop.icon
    }), prop.button && prop.button.children);
  })), /*#__PURE__*/_react.default.createElement("span", {
    className: "text-blueGray-500 mt-6 mb-3"
  }, leftCard.subtitle), leftCard.inputs.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Input.default, _extends({
      key: key
    }, prop));
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-3"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, leftCard.button)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "transition-all bg-blueGray-200 duration-700 ease-in-out h-full top-0 absolute w-1/2 left-0 transform " + rightCardTransform
  }, /*#__PURE__*/_react.default.createElement("form", {
    className: "flex items-center justify-center flex-col px-12 h-full text-center"
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: "text-2xp mb-2 font-bold"
  }, rightCard.title), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center"
  }, rightCard.socials.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      key: key
    }, prop.button), /*#__PURE__*/_react.default.createElement("i", {
      className: prop.icon
    }), prop.button && prop.button.children);
  })), /*#__PURE__*/_react.default.createElement("span", {
    className: "text-blueGray-500 mt-6 mb-3"
  }, rightCard.subtitle), rightCard.inputs.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Input.default, _extends({
      key: key
    }, prop));
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-3"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, rightCard.button)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "transition-all duration-700 ease-in-out transform absolute top-0 left-1/2 w-1/2 h-full overflow-hidden z-9999 " + panelTransform
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: panelColorClass[color] + " transition-all duration-700 ease-in-out transform relative -left-full h-full w-double " + panelInnerTransform
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "transition-all duration-700 ease-in-out transform absolute flex items-center justify-center flex-col p-10 text-center top-0 right-0 h-full w-1/2 " + panelLeftTransform
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xp mb-2 font-bold"
  }, detailsPanel.leftTitle), /*#__PURE__*/_react.default.createElement("p", {
    className: "mb-6"
  }, detailsPanel.leftDescription), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, detailsPanel.leftButton, {
    onClick: animation
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "transition-all duration-700 ease-in-out transform absolute flex items-center justify-center flex-col p-10 text-center top-0 h-full w-1/2 " + panelRightTransform
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xp mb-2 font-bold"
  }, detailsPanel.rightTitle), /*#__PURE__*/_react.default.createElement("p", {
    className: "mb-6"
  }, detailsPanel.rightDescription), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, detailsPanel.rightButton, {
    onClick: animation
  })))))));
}

var cardTypes = _propTypes.default.shape({
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object,
  // array of props to pass to the Input components
  inputs: _propTypes.default.arrayOf(_propTypes.default.object),
  socials: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.string,
    // props to pass to the Button component that wrapps the icon
    button: _propTypes.default.object
  }))
});

CardLoginRegister.defaultProps = {
  leftCard: {
    inputs: [],
    button: {},
    socials: []
  },
  rightCard: {
    inputs: [],
    button: {},
    socials: []
  },
  detailsPanel: {
    leftButton: {},
    rightButton: {}
  },
  color: "dark"
};
CardLoginRegister.propTypes = {
  leftCard: cardTypes,
  rightCard: cardTypes,
  detailsPanel: _propTypes.default.shape({
    leftTitle: _propTypes.default.string,
    leftDescription: _propTypes.default.string,
    leftButton: _propTypes.default.object,
    rightTitle: _propTypes.default.string,
    rightDescription: _propTypes.default.string,
    rightButton: _propTypes.default.object
  }),
  // color for the detailsPanel
  color: _propTypes.default.oneOf(["light", "dark", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
