// @notus-pro/react/CardProfileFullPage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProfileFullPage;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardProfileFullPage(_ref) {
  var image = _ref.image,
      name = _ref.name,
      location = _ref.location,
      description = _ref.description,
      showMore = _ref.showMore,
      stats = _ref.stats,
      buttons = _ref.buttons,
      achievements = _ref.achievements;
  var newDescription = [];

  if (!Array.isArray(description)) {
    newDescription = [description];
  } else {
    newDescription = description;
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-xl rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap justify-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full lg:w-3/12 px-4 lg:order-2 flex justify-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "shadow-xl rounded-full h-auto align-middle border-none absolute -m-16 -ml-20 lg:-ml-16 max-w-150-px"
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full lg:w-4/12 px-4 lg:order-3 lg:text-right lg:self-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-6 px-3 mt-32 sm:mt-0"
  }, buttons.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, prop, {
      key: key
    }));
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full lg:w-4/12 px-4 lg:order-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center py-4 lg:pt-4 pt-8"
  }, stats.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "mr-4 p-3 text-center"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "text-xl font-bold block tracking-wide text-blueGray-700"
    }, prop.title), /*#__PURE__*/_react.default.createElement("span", {
      className: "text-sm text-blueGray-500"
    }, prop.subtitle));
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mt-12"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-4xl font-bold leading-tight mb-2 mb-0"
  }, name), location ? /*#__PURE__*/_react.default.createElement("div", {
    className: "text-sm leading-normal mt-0 mb-2 text-blueGray-500 font-bold uppercase"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "fas fa-map-marker-alt mr-2 text-lg text-blueGray-500"
  }), location) : null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-2 text-blueGray-700 mt-10"
  }, achievements.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("p", {
      key: key,
      className: "mb-2 text-blueGray-700"
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: prop.icon + " mr-2 text-lg text-blueGray-500"
    }), prop.text);
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-10 py-10 border-t border-blueGray-300 text-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap justify-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full lg:w-9/12 px-4"
  }, newDescription.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("p", {
      key: key,
      className: "mb-4 text-lg leading-relaxed text-blueGray-800"
    }, prop);
  }), showMore && showMore.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, showMore, {
    className: "font-normal text-lightBlue-500 hover:text-lightBlue-600"
  })) : /*#__PURE__*/_react.default.createElement("a", _extends({}, showMore, {
    className: "font-normal text-lightBlue-500 hover:text-lightBlue-600"
  }))))))));
}

CardProfileFullPage.defaultProps = {
  stats: [],
  buttons: [],
  achievements: [],
  showMore: {}
};
CardProfileFullPage.propTypes = {
  image: _propTypes.default.string,
  name: _propTypes.default.string,
  location: _propTypes.default.string,
  // You send a simple string for a one paragraph description
  // or you can send an array of strings, that will generate
  // each string into a new paragraph
  description: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.arrayOf(_propTypes.default.string)]),
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  showMore: _propTypes.default.object,
  stats: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    subtitle: _propTypes.default.string
  })),
  // array of objects representing the props
  // that you can pass to the Button component
  buttons: _propTypes.default.arrayOf(_propTypes.default.object),
  achievements: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.string,
    text: _propTypes.default.string
  }))
};
