// @notus-pro/react/CardPricing.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardPricing;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

// components
function CardPricing(_ref) {
  var headText = _ref.headText,
      price = _ref.price,
      unit = _ref.unit,
      list = _ref.list,
      link = _ref.link;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-white mb-6 text-center shadow-lg rounded-lg relative flex flex-col min-w-0 break-words w-full mb-6 rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-transparent first:rounded-t px-5 py-3 border-b border-blueGray-200"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "font-bold my-2"
  }, headText)), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-5xl mt-0 mb-0 font-bold"
  }, price), /*#__PURE__*/_react.default.createElement("span", null, " ", unit, " "), /*#__PURE__*/_react.default.createElement("ul", {
    className: "mt-6 mb-0 list-none"
  }, list.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("li", {
      className: "py-1 text-blueGray-500",
      key: key
    }, /*#__PURE__*/_react.default.createElement("b", {
      className: "text-lightBlue-500"
    }, " ", prop.feature, " "), prop.text);
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-4 py-6 bg-transparent bg-transparent rounded-b px-4 py-3 border-t border-blueGray-200"
  }, link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, link, {
    className: "text-lightBlue-500"
  })) : /*#__PURE__*/_react.default.createElement("a", _extends({}, link, {
    className: "text-lightBlue-500"
  })))));
}

CardPricing.defaultProps = {
  list: []
};
CardPricing.propTypes = {
  headText: _propTypes.default.string,
  price: _propTypes.default.string,
  unit: _propTypes.default.string,
  list: _propTypes.default.arrayOf(_propTypes.default.shape({
    feature: _propTypes.default.string,
    text: _propTypes.default.string
  })),
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object
};
