// @notus-pro/react/CardPricingLogo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardPricingLogo;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardPricingLogo(_ref) {
  var description = _ref.description,
      title = _ref.title,
      list = _ref.list,
      line = _ref.line,
      image = _ref.image,
      button = _ref.button;
  var colors = {
    blueGray: "border-blueGray-500",
    red: "border-red-500",
    orange: "border-orange-500",
    amber: "border-amber-500",
    emerald: "border-emerald-500",
    teal: "border-teal-500",
    lightBlue: "border-lightBlue-500",
    indigo: "border-indigo-500",
    purple: "border-purple-500",
    pink: "border-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-12 shadow-lg rounded-lg bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "pt-4"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    className: "block mx-auto w-100-px",
    src: image
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center p-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "text-blueGray-500 mt-2 text-5xl"
  }, description), /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-2xl font-bold leading-tight mt-2 mb-2"
  }, title), /*#__PURE__*/_react.default.createElement("hr", {
    className: colors[line] + " my-2 mx-auto w-1/5"
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap"
  }, /*#__PURE__*/_react.default.createElement("ul", {
    className: "mt-2 mb-6 list-none mx-auto text-center flex flex-col"
  }, list.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("li", {
      key: key,
      className: "py-1 text-blueGray-500"
    }, prop);
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "pb-4"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button)))));
}

CardPricingLogo.defaultProps = {
  list: [],
  line: "blueGray",
  button: {}
};
CardPricingLogo.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object,
  list: _propTypes.default.arrayOf(_propTypes.default.string),
  line: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
