// @notus-pro/react/CardRegister.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardRegister;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Checkbox = _interopRequireDefault(require("./Checkbox.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardRegister(_ref) {
  var title = _ref.title,
      subtitle = _ref.subtitle,
      inputs = _ref.inputs,
      socials = _ref.socials,
      button = _ref.button,
      checkbox = _ref.checkbox,
      forgotPassword = _ref.forgotPassword,
      createAccount = _ref.createAccount;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-lg bg-blueGray-100"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "rounded-t-lg bg-blueGray-800 mb-0 px-6 py-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mb-3"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-blueGray-300 text-sm font-bold"
  }, title)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center"
  }, socials.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, prop.button, {
      key: key,
      color: prop.icon,
      fullWidth: false
    }), /*#__PURE__*/_react.default.createElement("i", {
      className: "mr-1 fab fa-" + prop.icon
    }), " ", prop.icon);
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-auto px-4 lg:px-10 py-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-blueGray-500 text-center mb-3 font-bold"
  }, /*#__PURE__*/_react.default.createElement("small", null, subtitle)), /*#__PURE__*/_react.default.createElement("form", null, inputs.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "relative w-full"
    }, /*#__PURE__*/_react.default.createElement("label", {
      className: "block uppercase text-blueGray-700 text-xs font-bold mb-2 ml-1"
    }, prop.label), /*#__PURE__*/_react.default.createElement(_Input.default, prop.input));
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-2 inline-block"
  }, /*#__PURE__*/_react.default.createElement(_Checkbox.default, checkbox)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mt-6"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button))))));
}

CardRegister.defaultProps = {
  inputs: [],
  button: {}
};
CardRegister.propTypes = {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  // It is represetnted by the props you
  // can pass to our Button component element
  button: _propTypes.default.object,
  socials: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.oneOf(["facebook", "twitter", "instagram", "github", "pinterest", "youtube", "vimeo", "slack", "dribbble", "reddit", "tumblr", "linkedin"]),
    // It is represetnted by the props you
    // can pass to our Button component element
    // for example you can add an onClick handler
    button: _propTypes.default.object
  })),
  // It is represetnted by the props you
  // can pass to our Checkbox component element
  checkbox: _propTypes.default.object,
  inputs: _propTypes.default.arrayOf(_propTypes.default.shape({
    label: _propTypes.default.string,
    // It is represetnted by the props you
    // can pass to our Input component element
    // NOTE: if you wish to generate a textarea, you will need to pass
    // // // inside this object >> type: "textarea"
    // // // full example >> input: { type: "textarea" }
    input: _propTypes.default.object
  }))
};
