// @notus-pro/react/CardInfoWithButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardInfoWithButton;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardInfoWithButton(_ref) {
  var icon = _ref.icon,
      color = _ref.color,
      title = _ref.title,
      description = _ref.description,
      button = _ref.button;
  var iconColor = {
    light: "bg-blueGray-200 text-blueGray-800",
    dark: "bg-blueGray-800 text-blueGray-100",
    blueGray: "bg-blueGray-500 text-white",
    red: "bg-red-500 text-white",
    orange: "bg-orange-500 text-white",
    amber: "bg-amber-500 text-white",
    emerald: "bg-emerald-500 text-white",
    teal: "bg-teal-500 text-white",
    lightBlue: "bg-lightBlue-500 text-white",
    indigo: "bg-indigo-500 text-white",
    purple: "bg-purple-500 text-white",
    pink: "bg-pink-500 text-white"
  };
  var textColor = {
    light: "text-blueGray-400",
    dark: "text-blueGray-800",
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "shadow-lg rounded-lg bg-white relative flex flex-col min-w-0 break-words bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-6 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: iconColor[color] + " shadow-lg rounded-full justify-center items-center mx-auto mb-6 inline-flex w-12 h-12"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: icon + " text-xl"
  })), /*#__PURE__*/_react.default.createElement("h6", {
    className: textColor[color] + " text-xl font-bold mt-0 mb-0"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-2 mb-4 leading-relaxed text-blueGray-500"
  }, description), /*#__PURE__*/_react.default.createElement(_Button.default, button))));
}

CardInfoWithButton.defaultProps = {
  color: "blueGray",
  button: {}
};
CardInfoWithButton.propTypes = {
  // icon name to be added on the <i></i> tag, for example "fa fa-heart"
  icon: _propTypes.default.string.isRequired,
  color: _propTypes.default.oneOf(["light", "dark", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  // if the background should have a gradient
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object
};
