// @notus-pro/react/CardProfileActions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProfileActions;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardProfileActions(_ref) {
  var headImage = _ref.headImage,
      image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      leftButton = _ref.leftButton,
      rightButton = _ref.rightButton,
      stats = _ref.stats;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "shadow-lg rounded-lg relative flex flex-col min-w-0 break-words bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-red-500 mb-12 w-full border-0 rounded-t-lg px-5 py-3 h-140-px bg-bottom bg-cover",
    style: {
      backgroundImage: "url(" + headImage + ")"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "rounded-full shadow-lg align-middle absolute mt-12 border-4 border-white max-w-150-px"
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center mt-8"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, leftButton), /*#__PURE__*/_react.default.createElement(_Button.default, rightButton)), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-4 flex justify-center"
  }, stats.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "text-center mr-4 p-4"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "font-bold text-xl block"
    }, prop.title), /*#__PURE__*/_react.default.createElement("span", {
      className: "text-blueGray-400"
    }, prop.subtitle));
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xl font-bold leading-tight mt-0 mb-1"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-lg text-blueGray-400"
  }, description)))));
}

CardProfileActions.defaultProps = {
  leftButton: {},
  rightButton: {},
  stats: []
};
CardProfileActions.propTypes = {
  headImage: _propTypes.default.string,
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the Button component
  leftButton: _propTypes.default.object,
  // props to pass to the Button component
  rightButton: _propTypes.default.object,
  stats: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    subtitle: _propTypes.default.string
  }))
};
