// @notus-pro/react/CardStatsMini.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardStatsMini;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function CardStatsMini(_ref) {
  var statSubtitle = _ref.statSubtitle,
      statTitle = _ref.statTitle,
      statArrow = _ref.statArrow,
      statPercent = _ref.statPercent,
      statPercentColor = _ref.statPercentColor,
      statDescripiron = _ref.statDescripiron,
      statIconName = _ref.statIconName,
      statIconColor = _ref.statIconColor;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words bg-white rounded-lg mb-6 xl:mb-0 shadow-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-auto p-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full pr-4 max-w-full flex-grow flex-1"
  }, /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-blueGray-400 uppercase font-bold text-xs"
  }, statSubtitle), /*#__PURE__*/_react.default.createElement("span", {
    className: "font-bold text-xl"
  }, statTitle)), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-auto pl-4 flex-initial"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full " + statIconColor
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: statIconName
  })))), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-sm text-blueGray-500 mt-4"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: statPercentColor + " mr-2"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: statArrow === "up" ? "fas fa-arrow-up" : statArrow === "down" ? "fas fa-arrow-down" : ""
  }), " ", statPercent, "%"), /*#__PURE__*/_react.default.createElement("span", {
    className: "whitespace-nowrap"
  }, statDescripiron)))));
}

CardStatsMini.defaultProps = {
  statArrow: "up"
};
CardStatsMini.propTypes = {
  statSubtitle: _propTypes.default.string,
  statTitle: _propTypes.default.string,
  statArrow: _propTypes.default.oneOf(["up", "down"]),
  statPercent: _propTypes.default.string,
  // can be any of the text color utilities
  // from tailwindcss
  statPercentColor: _propTypes.default.string,
  statDescripiron: _propTypes.default.string,
  statIconName: _propTypes.default.string,
  // can be any of the background color utilities
  // from tailwindcss
  statIconColor: _propTypes.default.string
};
