// @notus-pro/react/CardBlogIllustration.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardBlogImageHorizontal;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardBlogImageHorizontal(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      link = _ref.link;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-800 text-white shadow-lg rounded-lg overflow-hidden relative flex flex-col min-w-0 w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "w-full",
    style: {
      marginTop: "-45%"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-3xl mt-0 mb-2 my-4 font-bold"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-xl font-light"
  }, description), link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, link, {
    className: "font-semibold text-sm uppercase py-3 inline-block"
  }), "Read more") : /*#__PURE__*/_react.default.createElement("a", _extends({}, link, {
    className: "font-semibold text-sm uppercase py-3 inline-block"
  }), "Read more"))));
}

CardBlogImageHorizontal.defaultProps = {
  link: {}
};
CardBlogImageHorizontal.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // properties to pass to the link object after the description
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  // NOTE: you cannot pass a child component
  link: _propTypes.default.object
};
