// @notus-pro/react/CardProfileFullDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProfileFullDetails;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _ImagesOverlap = _interopRequireDefault(require("./ImagesOverlap.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardProfileFullDetails(_ref) {
  var image = _ref.image,
      details = _ref.details,
      stats = _ref.stats,
      team = _ref.team,
      button = _ref.button;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left overflow-hidden shadow-lg rounded-lg relative flex flex-col md:flex-row min-w-0 break-words bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    className: "w-full md:w-1/2",
    src: image
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full mt-4 flex flex-col items-center md:w-1/2 md:mt-0"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-10 flex-auto w-full"
  }, details.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "py-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "text-left p-0"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "text-blueGray-400 uppercase text-sm font-bold"
    }, prop.subtitle), /*#__PURE__*/_react.default.createElement("span", {
      className: "font-bold block text-2xl"
    }, prop.title)));
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "py-2 flex border-t border-blueGray-200"
  }, stats.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "text-center mr-4 pr-4 py-4"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "font-bold block uppercase text-lg"
    }, prop.title), /*#__PURE__*/_react.default.createElement("span", {
      className: "text-blueGray-500"
    }, prop.subtitle));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "py-4"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "font-bold block uppercase mb-2"
  }, team.title), /*#__PURE__*/_react.default.createElement("div", {
    className: "pl-4"
  }, /*#__PURE__*/_react.default.createElement(_ImagesOverlap.default, team.imagesOverlap))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full bg-transparent rounded-b px-4 pt-6 border-t border-blueGray-200 text-center mt-4"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button))))));
}

CardProfileFullDetails.defaultProps = {
  details: [],
  stats: [],
  team: {},
  button: {}
};
CardProfileFullDetails.propTypes = {
  image: _propTypes.default.string,
  details: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    subtitle: _propTypes.default.string
  })),
  stats: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    subtitle: _propTypes.default.string
  })),
  team: _propTypes.default.shape({
    title: _propTypes.default.string,
    // props to pass to the ImagesOverlap component
    imagesOverlap: _propTypes.default.object
  }),
  // props to pass to the Button component
  button: _propTypes.default.object
};
