// @notus-pro/react/CardPricingDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardPricingDetails;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

// components
function CardPricingDetails(_ref) {
  var head = _ref.head,
      title = _ref.title,
      description = _ref.description,
      list = _ref.list,
      link = _ref.link,
      footer = _ref.footer,
      gradient = _ref.gradient,
      color = _ref.color;
  var listColors = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  var colors = {
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500",
    "blueGray-gradient": "bg-gradient-to-tl from-blueGray-700 to-blueGray-500 bg-blueGray-500",
    "red-gradient": "bg-gradient-to-tl from-red-700 to-red-500 bg-red-500",
    "orange-gradient": "bg-gradient-to-tl from-orange-700 to-orange-500 bg-orange-500",
    "amber-gradient": "bg-gradient-to-tl from-amber-700 to-amber-500 bg-amber-500",
    "emerald-gradient": "bg-gradient-to-tl from-emerald-700 to-emerald-500 bg-emerald-500",
    "teal-gradient": "bg-gradient-to-tl from-teal-700 to-teal-500 bg-teal-500",
    "lightBlue-gradient": "bg-gradient-to-tl from-lightBlue-700 to-lightBlue-500 bg-lightBlue-500",
    "indigo-gradient": "bg-gradient-to-tl from-indigo-700 to-indigo-500 bg-indigo-500",
    "purple-gradient": "bg-gradient-to-tl from-purple-700 to-purple-500 bg-purple-500",
    "pink-gradient": "bg-gradient-to-tl from-pink-700 to-pink-500 bg-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: colors[color + (gradient ? "-gradient" : "")] + " mb-6 text-center text-white shadow-lg rounded-lg relative flex flex-col min-w-0 break-words w-full mb-6 text-white"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-transparent first:rounded-lg-t px-5 py-3 mx-4 border-b border-black border-opacity-10"
  }, /*#__PURE__*/_react.default.createElement("h4", {
    className: "font-bold uppercase my-2 text-white"
  }, head)), /*#__PURE__*/_react.default.createElement("div", {
    className: "lg:px-12 px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-6xl font-normal leading-tight font-bold text-white"
  }, title), /*#__PURE__*/_react.default.createElement("span", null, description), /*#__PURE__*/_react.default.createElement("ul", {
    className: "my-8 list-none"
  }, list.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("li", {
      key: key,
      className: "py-2"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex items-center"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: listColors[prop.color] + " inline-flex text-center rounded-full mr-3 items-center justify-center w-10 h-10 bg-white"
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: prop.icon
    })), /*#__PURE__*/_react.default.createElement("h6", {
      className: "text-lg"
    }, prop.text)));
  })), footer && footer.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, link, {
    className: "py-2 text-white uppercase font-bold"
  })), footer && footer.to === undefined && /*#__PURE__*/_react.default.createElement("a", _extends({}, link, {
    className: "py-2 text-white uppercase font-bold"
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-2 px-4 py-4 border-t border-black border-opacity-10 mx-4"
  }, footer && footer.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, footer, {
    className: "text-white opacity-75"
  })), footer && footer.to === undefined && /*#__PURE__*/_react.default.createElement("a", _extends({}, footer, {
    className: "text-white opacity-75"
  })))));
}

CardPricingDetails.defaultProps = {
  link: {},
  footer: {},
  list: [],
  gradient: false
};
CardPricingDetails.propTypes = {
  head: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object,
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  footer: _propTypes.default.object,
  list: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.string,
    text: _propTypes.default.string,
    color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
  })),
  gradient: _propTypes.default.bool,
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
