// @notus-pro/react/CardResetPassword.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardResetPassword;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardResetPassword(_ref) {
  var image = _ref.image,
      title = _ref.title,
      subtitle = _ref.subtitle,
      input = _ref.input,
      button = _ref.button;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-white relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "pb-6 rounded-t-lg overflow-hidden"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: image,
    alt: "...",
    className: "w-full rounded-t-lg",
    style: {
      marginTop: "-46%"
    }
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "lg:p-10 px-4 py-4 flex-auto rounded-b-lg"
  }, /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-2xl font-bold text-center"
  }, title), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center text-blueGray-500 mb-5 mt-1"
  }, /*#__PURE__*/_react.default.createElement("p", null, subtitle)), /*#__PURE__*/_react.default.createElement(_Input.default, input), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mt-6"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button)))));
}

CardResetPassword.defaultProps = {
  input: {},
  button: {}
};
CardResetPassword.propTypes = {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  image: _propTypes.default.string,
  // It is represetnted by the props you
  // can pass to our Input component element
  input: _propTypes.default.object,
  // It is represetnted by the props you
  // can pass to our Button component element
  button: _propTypes.default.object
};
