// @notus-pro/react/CardBlogandAuthor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardBlogandAuthor;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardBlogandAuthor(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      user = _ref.user,
      icon = _ref.icon,
      text = _ref.text,
      link = _ref.link;

  var linkContent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    className: "shadow-lg rounded-lg w-full flex-shrink-0",
    src: image
  }), /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xl font-bold leading-tight ml-1 mt-6"
  }, title));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12 relative flex flex-col min-w-0 break-words bg-white w-full mb-6 rounded-lg"
  }, link && link.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, link, linkContent), link && link.to === undefined && /*#__PURE__*/_react.default.createElement("a", link, linkContent), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-3 leading-relaxed text-blueGray-500 ml-1"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-6 flex items-center"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: user.image,
    className: "rounded-full mr-3 max-w-40-px"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between items-center"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "font-bold mt-0 mb-0"
  }, user.name, /*#__PURE__*/_react.default.createElement("small", {
    className: "text-sm text-blueGray-500 float-right"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: icon + " opacity-75 text-red-500 mr-1"
  }), " ", text)))))));
}

CardBlogandAuthor.defaultProps = {
  user: {},
  link: {}
};
CardBlogandAuthor.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  user: _propTypes.default.shape({
    image: _propTypes.default.string,
    name: _propTypes.default.string
  }),
  icon: _propTypes.default.string,
  text: _propTypes.default.string,
  // wrapper link for the image and title
  // if you send a prop named to the link
  // will be generated as a Link from react-router-dom
  // otherwise it will be generated as an <a> tag
  link: _propTypes.default.object
};
