// @notus-pro/react/CardFullTable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardFullTable;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _DropdownButton = _interopRequireDefault(require("./DropdownButton.js"));

var _Progress = _interopRequireDefault(require("./Progress.js"));

var _ImagesOverlap = _interopRequireDefault(require("./ImagesOverlap.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable jsx-a11y/anchor-has-content */
// components
function CardFullTable(_ref) {
  var title = _ref.title,
      head = _ref.head,
      body = _ref.body,
      color = _ref.color,
      children = _ref.children;
  var cardColors = {
    white: "bg-white text-blueGray-700",
    light: "bg-blueGray-200 text-blueGray-700",
    blueGray: "bg-blueGray-900 text-white",
    red: "bg-red-900 text-white",
    orange: "bg-orange-900 text-white",
    amber: "bg-amber-900 text-white",
    emerald: "bg-emerald-900 text-white",
    teal: "bg-teal-900 text-white",
    lightBlue: "bg-lightBlue-900 text-white",
    indigo: "bg-indigo-900 text-white",
    purple: "bg-purple-900 text-white",
    pink: "bg-pink-900 text-white"
  };
  var headColors = {
    white: "bg-blueGray-100 text-blueGray-500 border-blueGray-200",
    light: "bg-blueGray-300 text-blueGray-700 border-blueGray-400",
    blueGray: "bg-blueGray-800 text-blueGray-300 border-blueGray-700",
    red: "bg-red-800 text-red-300 border-red-700",
    orange: "bg-orange-800 text-orange-300 border-orange-700",
    amber: "bg-amber-800 text-amber-300 border-amber-700",
    emerald: "bg-emerald-800 text-emerald-300 border-emerald-700",
    teal: "bg-teal-800 text-teal-300 border-teal-700",
    lightBlue: "bg-lightBlue-800 text-lightBlue-300 border-lightBlue-700",
    indigo: "bg-indigo-800 text-indigo-300 border-indigo-700",
    purple: "bg-purple-800 text-purple-300 border-purple-700",
    pink: "bg-pink-800 text-pink-300 border-pink-700"
  };
  var titleColors = {
    white: "text-blueGray-700",
    light: "text-blueGray-700",
    blueGray: "text-white",
    red: "text-white",
    orange: "text-white",
    amber: "text-white",
    emerald: "text-white",
    teal: "text-white",
    lightBlue: "text-white",
    indigo: "text-white",
    purple: "text-white",
    pink: "text-white"
  };
  var imageTextColors = {
    white: "text-blueGray-700",
    light: "text-blueGray-700",
    blueGray: "text-white",
    red: "text-white",
    orange: "text-white",
    amber: "text-white",
    emerald: "text-white",
    teal: "text-white",
    lightBlue: "text-white",
    indigo: "text-white",
    purple: "text-white",
    pink: "text-white"
  };
  var iconColors = {
    white: "text-white",
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words w-full mb-8 shadow-lg rounded-lg " + cardColors[color]
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-6 py-4 border-0"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap items-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full max-w-full flex-grow flex-1"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "font-bold text-lg " + titleColors[color]
  }, title)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "block w-full overflow-x-auto"
  }, /*#__PURE__*/_react.default.createElement("table", {
    className: "items-center w-full bg-transparent border-collapse"
  }, /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, head.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("th", {
      key: key,
      className: "px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-bold text-left " + headColors[color]
    }, prop);
  }))), /*#__PURE__*/_react.default.createElement("tbody", null, body.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      key: key
    }, prop.map(function (colProp, colKey) {
      if (typeof colProp === "string") {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, colProp));
      } else if (colProp.bold) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, /*#__PURE__*/_react.default.createElement("span", {
          className: "ml-3 font-bold " + +imageTextColors[color]
        }, colProp.bold)));
      } else if (colProp.arrow) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, /*#__PURE__*/_react.default.createElement("i", {
          className: "fas fa-arrow-" + colProp.arrow + " mr-2 " + iconColors[colProp.color]
        }), colProp.text));
      } else if (colProp.images) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, /*#__PURE__*/_react.default.createElement(_ImagesOverlap.default, colProp.images)));
      } else if (colProp.progress) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4 min-w-140-px"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, /*#__PURE__*/_react.default.createElement("span", {
          className: "mr-2"
        }, colProp.text), /*#__PURE__*/_react.default.createElement(_Progress.default, colProp.progress)));
      } else if (colProp.color) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, /*#__PURE__*/_react.default.createElement("i", {
          className: "fas fa-circle mr-2 " + iconColors[colProp.color]
        }), colProp.text));
      } else if (colProp.image) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, /*#__PURE__*/_react.default.createElement("img", {
          src: colProp.image,
          className: "h-12 w-12 bg-white rounded-full border p-1",
          alt: "..."
        }), " ", /*#__PURE__*/_react.default.createElement("span", {
          className: "ml-3 font-bold " + +imageTextColors[color]
        }, colProp.text)));
      } else if (colProp.dropdown) {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4 pb-6 pl-6 text-right"
        }, /*#__PURE__*/_react.default.createElement(_DropdownButton.default, colProp.dropdown));
      } else {
        return /*#__PURE__*/_react.default.createElement("td", {
          key: colKey,
          className: "border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "flex items-center"
        }, colProp));
      }
    }));
  })))), children && /*#__PURE__*/_react.default.createElement("div", {
    className: "rounded-b mt-0 px-4 py-3 border-0"
  }, children)));
}

CardFullTable.defaultProps = {
  head: [],
  body: [],
  children: null,
  color: "white"
};
CardFullTable.propTypes = {
  title: _propTypes.default.string,
  head: _propTypes.default.arrayOf(_propTypes.default.string),
  color: _propTypes.default.oneOf(["white", "light", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  body: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.oneOfType([// simple string
  _propTypes.default.string, // bold text
  _propTypes.default.shape({
    bold: _propTypes.default.string
  }), // image and text
  _propTypes.default.shape({
    image: _propTypes.default.string,
    text: _propTypes.default.string
  }), // round icon and text
  _propTypes.default.shape({
    text: _propTypes.default.string,
    color: _propTypes.default.oneOf(["white", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
  }), // arrow icon and text
  _propTypes.default.shape({
    text: _propTypes.default.string,
    arrow: _propTypes.default.oneOf(["up", "down"]),
    color: _propTypes.default.oneOf(["white", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
  }), _propTypes.default.shape({
    // props to pass to the ImagesOverlap component
    images: _propTypes.default.object
  }), _propTypes.default.shape({
    text: _propTypes.default.string,
    // props to pass to the Progress component
    progress: _propTypes.default.object
  }), _propTypes.default.shape({
    // props to pass to the DropdownButton component
    dropdown: _propTypes.default.object
  }), // if you want to pass your own component
  // NOTE: your component might break this Card component
  _propTypes.default.node]))),
  // this will appear at the bottom of the Table, for example,
  // you can use this to make a pagination component
  children: _propTypes.default.node
};
