// @notus-pro/react/CardInfoImage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardInfoImage;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardInfoImage(_ref) {
  var title = _ref.title,
      description = _ref.description,
      image = _ref.image,
      color = _ref.color;
  var bgColor = {
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500"
  };
  var textColor = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "shadow-lg rounded border-0 relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg text-white " + bgColor[color]
  }, /*#__PURE__*/_react.default.createElement("img", {
    className: "w-full flex-shrink-0 rounded-t-lg",
    src: image,
    alt: "..."
  }), /*#__PURE__*/_react.default.createElement("blockquote", {
    className: "p-6 relative mb-4"
  }, /*#__PURE__*/_react.default.createElement("svg", {
    preserveAspectRatio: "none",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 583 95",
    className: "block w-full absolute text-blueGray-700 h-95-px -top-94-px left-0"
  }, /*#__PURE__*/_react.default.createElement("polygon", {
    points: "0,52 583,95 0,95",
    className: textColor[color] + " fill-current"
  }), /*#__PURE__*/_react.default.createElement("polygon", {
    points: "0,42 583,95 683,0 0,95",
    opacity: ".2",
    className: textColor[color] + " fill-current"
  })), /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-2xl font-semibold mt-0 text-white"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-white mt-2"
  }, description))));
}

CardInfoImage.defaultProps = {
  color: "blueGray"
};
CardInfoImage.propTypes = {
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  image: _propTypes.default.string,
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
