// @notus-pro/react/CardChatPeople.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardChatPeople;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Contact = _interopRequireDefault(require("./Contact.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardChatPeople(_ref) {
  var input = _ref.input,
      contacts = _ref.contacts;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center bg-white border-0 shadow-lg rounded-lg relative flex flex-col min-w-0 break-words w-full mb-6"
  }, input && /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-200 rounded-t-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "my-4 mb-3 px-3 relative"
  }, /*#__PURE__*/_react.default.createElement(_Input.default, _extends({}, input, {
    border: "borderless",
    leftIcon: "fas fa-search"
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-4 flex pl-0 mb-3 flex-col"
  }, contacts.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Contact.default, _extends({
      key: key
    }, prop));
  }))));
}

CardChatPeople.defaultProps = {
  contacts: []
};
CardChatPeople.propTypes = {
  // props to pass to the Input component
  input: _propTypes.default.object,
  // array of props to pass to the Contact component
  contacts: _propTypes.default.arrayOf(_propTypes.default.object)
};
