// @notus-pro/react/CardDetailSettingsUser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardDetailSettingsUser;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames2 = _interopRequireDefault(require("classnames"));

var _ImageUpload = _interopRequireDefault(require("./ImageUpload.js"));

var _Progress = _interopRequireDefault(require("./Progress.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Select = _interopRequireDefault(require("./Select.js"));

var _Checkbox = _interopRequireDefault(require("./Checkbox.js"));

var _Radio = _interopRequireDefault(require("./Radio.js"));

var _Alert = _interopRequireDefault(require("./Alert.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function CardDetailSettingsUser(_ref) {
  var imageUpload = _ref.imageUpload,
      userName = _ref.userName,
      progress = _ref.progress,
      saveButton = _ref.saveButton,
      cancelButton = _ref.cancelButton,
      defaultOpened = _ref.defaultOpened,
      tabs = _ref.tabs,
      tabsColor = _ref.tabsColor,
      onTabsChange = _ref.onTabsChange;

  var _React$useState = _react.default.useState(defaultOpened),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(false),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      oldInTransition = _React$useState4[0],
      setOldInTransition = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      newInTransition = _React$useState6[0],
      setNewInTransition = _React$useState6[1];

  var toggleNew = function toggleNew(e, newOpen, text) {
    e.preventDefault();

    if (!newInTransition && !oldInTransition) {
      setOldInTransition(true);
      setTimeout(function () {
        setOpen(newOpen);
      }, 500);
      setTimeout(function () {
        setOldInTransition(false);
        setNewInTransition(true);
      }, 600);
      setTimeout(function () {
        setNewInTransition(false);
      }, 1100);
    }

    onTabsChange(newOpen, text);
  };

  var widths = {
    1: "lg:w-1/12",
    2: "lg:w-2/12",
    3: "lg:w-3/12",
    4: "lg:w-4/12",
    5: "lg:w-5/12",
    6: "lg:w-6/12",
    7: "lg:w-7/12",
    8: "lg:w-8/12",
    9: "lg:w-9/12",
    10: "lg:w-10/12",
    11: "lg:w-11/12",
    12: "lg:w-12/12"
  };
  var anchorActiveColors = {
    blueGray: "text-blueGray-500 hover:text-blueGray-700",
    red: "text-red-500 hover:text-red-700",
    orange: "text-orange-500 hover:text-orange-700",
    amber: "text-amber-500 hover:text-amber-700",
    emerald: "text-emerald-500 hover:text-emerald-700",
    teal: "text-teal-500 hover:text-teal-700",
    lightBlue: "text-lightBlue-500 hover:text-lightBlue-700",
    indigo: "text-indigo-500 hover:text-indigo-700",
    purple: "text-purple-500 hover:text-purple-700",
    pink: "text-pink-500 hover:text-pink-700"
  };
  var anchorNonActiveColors = {
    blueGray: "text-blueGray-400 hover:text-blueGray-700",
    red: "text-blueGray-400 hover:text-red-700",
    orange: "text-blueGray-400 hover:text-orange-700",
    amber: "text-blueGray-400 hover:text-amber-700",
    emerald: "text-blueGray-400 hover:text-emerald-700",
    teal: "text-blueGray-400 hover:text-teal-700",
    lightBlue: "text-blueGray-400 hover:text-lightBlue-700",
    indigo: "text-blueGray-400 hover:text-indigo-700",
    purple: "text-blueGray-400 hover:text-purple-700",
    pink: "text-blueGray-400 hover:text-pink-700"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-white mb-0 relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full px-4 mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-3/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative py-12"
  }, /*#__PURE__*/_react.default.createElement("section", {
    className: "text-center block"
  }, /*#__PURE__*/_react.default.createElement(_ImageUpload.default, imageUpload), /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-3xl font-semibold mt-2 mb-4"
  }, userName)), /*#__PURE__*/_react.default.createElement("section", {
    className: "block"
  }, /*#__PURE__*/_react.default.createElement("ul", {
    className: "flex-col flex flex-wrap list-none pl-0 mb-0"
  }, tabs.map(function (prop, key) {
    var _classnames;

    return /*#__PURE__*/_react.default.createElement("li", {
      key: key,
      className: "last:border-0 border-b border-solid py-1"
    }, /*#__PURE__*/_react.default.createElement("a", {
      className: (0, _classnames2.default)("text-blueGray-700 text-xs uppercase py-3 pl-2 font-bold block duration-300 transition-all ease-in-out", (_classnames = {}, _defineProperty(_classnames, anchorNonActiveColors[tabsColor], key !== open), _defineProperty(_classnames, anchorActiveColors[tabsColor], key === open), _classnames)),
      onClick: function onClick(e) {
        return toggleNew(e, key, prop.tabName);
      },
      href: "#pablo"
    }, prop.tabName));
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/_react.default.createElement(_Progress.default, progress)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "ml-auto px-4 relative w-full md:w-8/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative py-12"
  }, tabs.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: (0, _classnames2.default)("my-8 duration-500 transition-all ease-in-out", {
        hidden: key !== open,
        block: key === open,
        "opacity-0": key === open && oldInTransition,
        "opacity-100": key === open && newInTransition
      })
    }, prop.tabContent && prop.tabContent.map(function (contentProp, contentKey) {
      if (contentProp.table) {
        return /*#__PURE__*/_react.default.createElement("div", {
          key: contentKey,
          className: "mb-6"
        }, /*#__PURE__*/_react.default.createElement("header", {
          className: "block relative"
        }, /*#__PURE__*/_react.default.createElement("h2", {
          className: "text-2xl font-semibold mt-0 mb-2"
        }, contentProp.title)), /*#__PURE__*/_react.default.createElement("hr", {
          className: "w-full mt-2 mb-8 h-0 border-blueGray-400"
        }), /*#__PURE__*/_react.default.createElement("table", {
          className: "w-full mb-4 text-blueGray-800 border-collapse items-center"
        }, /*#__PURE__*/_react.default.createElement("thead", {
          className: "items-center"
        }, /*#__PURE__*/_react.default.createElement("tr", {
          className: "text-left"
        }, contentProp.table.head.map(function (tHeadProp, tHeadKey) {
          return /*#__PURE__*/_react.default.createElement("th", {
            key: tHeadKey,
            className: "align-middle uppercase font-bold text-sm p-3 border-t text-blueGray-500"
          }, /*#__PURE__*/_react.default.createElement("small", null, tHeadProp));
        }))), /*#__PURE__*/_react.default.createElement("tbody", {
          className: ""
        }, contentProp.table.body.map(function (tBodyProp, tBodyKey) {
          return /*#__PURE__*/_react.default.createElement("tr", {
            key: tBodyKey
          }, tBodyProp.map(function (colProp, colKey) {
            if (colProp.image) {
              return /*#__PURE__*/_react.default.createElement("td", {
                key: colKey,
                className: "align-middle uppercase font-bold text-sm p-3 border-t text-blueGray-500"
              }, /*#__PURE__*/_react.default.createElement("img", {
                alt: "...",
                className: "bg-white inline-flex justify-center items-center w-12",
                src: colProp.image
              }));
            } else if (colProp.radio) {
              return /*#__PURE__*/_react.default.createElement("td", {
                key: colKey,
                className: "align-middle uppercase font-normal p-3 border-t"
              }, /*#__PURE__*/_react.default.createElement(_Radio.default, colProp.radio));
            } else if (colProp.button) {
              return /*#__PURE__*/_react.default.createElement("td", {
                key: colKey,
                className: "align-middle uppercase font-normal p-3 border-t"
              }, /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
                key: colKey
              }, colProp.button)));
            } else {
              return /*#__PURE__*/_react.default.createElement("td", {
                key: colKey,
                className: "align-middle uppercase font-normal p-3 border-t"
              }, colProp);
            }
          }));
        }))));
      } else if (contentProp.inputs) {
        return /*#__PURE__*/_react.default.createElement("div", {
          key: contentKey,
          className: "mb-6"
        }, /*#__PURE__*/_react.default.createElement("header", {
          className: "block relative"
        }, /*#__PURE__*/_react.default.createElement("h2", {
          className: "text-2xl font-semibold mt-0 mb-2"
        }, contentProp.title)), /*#__PURE__*/_react.default.createElement("hr", {
          className: "w-full mt-2 mb-8 h-0 border-blueGray-400"
        }), contentProp.inputs && contentProp.inputs.map(function (inputProp, inputKey) {
          var label = inputProp.label;
          var element;

          if (inputProp.input) {
            element = /*#__PURE__*/_react.default.createElement(_Input.default, inputProp.input);
          } else {
            element = /*#__PURE__*/_react.default.createElement(_Select.default, inputProp.select);
          }

          return /*#__PURE__*/_react.default.createElement("div", {
            key: inputKey,
            className: "flex flex-wrap -mx-4"
          }, /*#__PURE__*/_react.default.createElement("div", {
            className: "self-center px-4 relative w-full md:w-3/12 text-right"
          }, /*#__PURE__*/_react.default.createElement("label", {
            className: "block uppercase text-blueGray-700 text-xs font-bold mb-2 ml-1"
          }, label)), /*#__PURE__*/_react.default.createElement("div", {
            className: "self-center px-4 relative w-full md:w-9/12"
          }, /*#__PURE__*/_react.default.createElement("div", {
            className: "flex flex-wrap -mx-4"
          }, /*#__PURE__*/_react.default.createElement("div", {
            className: "px-4 relative w-full " + widths[inputProp.width]
          }, element))));
        }));
      } else if (contentProp.list) {
        return /*#__PURE__*/_react.default.createElement("div", {
          key: contentKey,
          className: "mb-6"
        }, /*#__PURE__*/_react.default.createElement("header", {
          className: "block relative"
        }, /*#__PURE__*/_react.default.createElement("h2", {
          className: "text-2xl font-semibold mt-0 mb-2"
        }, contentProp.title)), /*#__PURE__*/_react.default.createElement("hr", {
          className: "w-full mt-2 mb-8 h-0 border-blueGray-400"
        }), contentProp.list && contentProp.list.map(function (listProp, listKey) {
          var text = /*#__PURE__*/_react.default.createElement("span", null, listProp.text);

          var element;

          if (listProp.checkbox) {
            element = /*#__PURE__*/_react.default.createElement(_Checkbox.default, listProp.checkbox);
          } else {
            element = /*#__PURE__*/_react.default.createElement(_Radio.default, listProp.radio);
          }

          return /*#__PURE__*/_react.default.createElement("div", {
            key: listKey,
            className: "flex items-center justify-between mb-3 pt-0 relative"
          }, listProp.inverted ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, element, text) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, text, element));
        }));
      } else {
        return /*#__PURE__*/_react.default.createElement(_Alert.default, _extends({
          key: contentKey
        }, contentProp.alert));
      }
    }));
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12 justify-center flex flex-wrap w-full"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, cancelButton), /*#__PURE__*/_react.default.createElement(_Button.default, saveButton))))));
}

var inputsObject = {
  // NOTE: this width only applies on large devices
  width: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
  // if you wish, you can send somthing like
  // <span className="mr-2">Label Name</span><span className="text-red-500">*</span>
  // NOTE: the label tag will be auto generated
  label: _propTypes.default.node
};

var inputsShape = _propTypes.default.shape(_objectSpread(_objectSpread({}, inputsObject), {}, {
  // props to pass to the Input component
  input: _propTypes.default.object
}));

var selectsShape = _propTypes.default.shape(_objectSpread(_objectSpread({}, inputsObject), {}, {
  // props to pass to the Select component
  select: _propTypes.default.object
}));

var inputsTypes = _propTypes.default.arrayOf(_propTypes.default.oneOfType([inputsShape, selectsShape]));

var listsObject = {
  text: _propTypes.default.string,
  // if you want the text to appear last or not
  inverted: _propTypes.default.bool
};

var listsCheckboxShape = _propTypes.default.shape(_objectSpread(_objectSpread({}, listsObject), {}, {
  // props to pass to the Checkbox component
  checkbox: _propTypes.default.object
}));

var listsRadioShape = _propTypes.default.shape(_objectSpread(_objectSpread({}, listsObject), {}, {
  // props to pass to the Radio component
  radio: _propTypes.default.object
}));

var listsTypes = _propTypes.default.arrayOf(_propTypes.default.oneOfType([listsCheckboxShape, listsRadioShape]));

CardDetailSettingsUser.defaultProps = {
  imageUpload: {},
  progress: {},
  defaultOpened: 0,
  onTabsChange: function onTabsChange() {},
  tabsColor: "blueGray",
  tabs: [],
  saveButton: {},
  cancelButton: {}
};
CardDetailSettingsUser.propTypes = {
  // props to pass to the ImageUpload component
  imageUpload: _propTypes.default.object,
  userName: _propTypes.default.string,
  // props to pass to the Progress component
  progress: _propTypes.default.object,
  // props to pass to the Button component
  saveButton: _propTypes.default.object,
  // props to pass to the Button component
  cancelButton: _propTypes.default.object,
  // 0 represents the first element
  // also, you should note that
  // the number should not be lower then 0
  // or higher than the number of tabs - 1
  defaultOpened: _propTypes.default.number,
  // this function has two parameters,
  // index number of the tab that was pressed
  // and the tabName of that tab
  // example: onTabsChange={(index,name) => console.log(index,name)}
  onTabsChange: _propTypes.default.func,
  tabsColor: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  tabs: _propTypes.default.arrayOf(_propTypes.default.shape({
    tabName: _propTypes.default.string,
    tabContent: _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.shape({
      title: _propTypes.default.string,
      inputs: inputsTypes
    }), _propTypes.default.shape({
      // props that you can pass to the Alert component
      alert: _propTypes.default.object
    }), _propTypes.default.shape({
      title: _propTypes.default.string,
      list: listsTypes
    }), _propTypes.default.shape({
      title: _propTypes.default.string,
      table: _propTypes.default.shape({
        head: _propTypes.default.arrayOf(_propTypes.default.string),
        body: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.shape({
          image: _propTypes.default.string
        }), // props to pass to the Radio component
        _propTypes.default.shape({
          radio: _propTypes.default.object
        }), // props to pass to the Button component
        _propTypes.default.shape({
          button: _propTypes.default.object
        })])))
      })
    })]))
  }))
};
