// @notus-pro/react/CardBlogFullBg.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardBlogFullBg;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable react/jsx-no-target-blank */
// components
function CardBlogFullBg(_ref) {
  var image = _ref.image,
      title = _ref.title,
      subtitle = _ref.subtitle,
      link = _ref.link,
      innerLink = _ref.innerLink,
      rounded = _ref.rounded,
      size = _ref.size;
  var sizes = {
    sm: "h-330-px",
    md: "h-430-px",
    lg: "h-530-px"
  };

  var body = /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute text-left p-6 bottom-0"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-xl leading-normal mb-0 text-white opacity-75"
  }, subtitle), /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-2xl font-bold leading-tight mt-0 mb-2 text-white"
  }, title));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: sizes[size] + " overflow-hidden relative flex flex-col min-w-0 break-words bg-white w-full shadow-lg transition-all duration-150 ease-in-out hover:z-1 hover:transform hover:scale-110 group" + (rounded ? " rounded-lg mb-4" : "")
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-50-center bg-cover transition-all duration-1000 ease-in-out group-hover:transform group-hover:scale-110" + (rounded ? " rounded-lg" : ""),
    style: {
      backgroundImage: "url('" + image + "')",
      backfaceVisibility: "hidden"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-black opacity-50" + (rounded ? " rounded-lg" : "")
  }), link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, link, body) : /*#__PURE__*/_react.default.createElement("a", link, body)));
}

CardBlogFullBg.defaultProps = {
  innerLink: false,
  rounded: true,
  size: "md"
};
CardBlogFullBg.propTypes = {
  image: _propTypes.default.string.isRequired,
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  // if it should have rounded coreners or not
  rounded: _propTypes.default.bool,
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object,
  // this is for the height
  size: _propTypes.default.oneOf(["sm", "lg", "md"])
};
