// @notus-pro/react/CardTableDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardTableDetails;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardTableDetails(_ref) {
  var title = _ref.title,
      description = _ref.description,
      table = _ref.table,
      link = _ref.link;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-white text-center shadow-lg rounded-lg relative flex flex-col min-w-0 break-words bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-transparent first:rounded-t px-5 py-3 border-b border-blueGray-300"
  }, /*#__PURE__*/_react.default.createElement("h4", {
    className: "pt-2 text-2xl font-bold mt-0 mb-2 text-lightBlue-500"
  }, title)), /*#__PURE__*/_react.default.createElement("div", {
    className: "lg:px-12 px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "leading-relaxed text-blueGray-500"
  }, description), /*#__PURE__*/_react.default.createElement("table", {
    className: "mt-6 w-full mb-4 text-blueGray-700 border-collapse border border-solid border-blueGray-200"
  }, /*#__PURE__*/_react.default.createElement("tbody", null, table.map(function (row, key) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      key: key
    }, row.map(function (col, colKey) {
      return /*#__PURE__*/_react.default.createElement("td", {
        key: colKey,
        className: "text-center p-3 border-t border"
      }, col);
    }));
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-transparent mt-4 py-6 bg-transparent rounded-b px-4 py-3 border-t border-blueGray-300"
  }, link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({
    className: "text-lightBlue-500"
  }, link)) : /*#__PURE__*/_react.default.createElement("a", _extends({
    className: "text-lightBlue-500"
  }, link)))));
}

CardTableDetails.defaultProps = {
  table: [],
  link: {}
};
CardTableDetails.propTypes = {
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // example: [["one","two"],["three","four","five"]]
  // will be generated as a table with two rows,
  // the first row with two columns -         one | two
  // the second row with three columns - three | four | five
  table: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.string)),
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object
};
