// @notus-pro/react/CardProjectTeam.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProjectTeam;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ImagesOverlap = _interopRequireDefault(require("./ImagesOverlap.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardProjectTeam(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      icon = _ref.icon,
      imagesOverlap = _ref.imagesOverlap;
  var colors = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-6 text-center shadow-lg rounded-lg relative flex flex-col bg-white p-6 w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center justify-between pb-4 pt-2 border-b border-blueGray-300"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "float-left ml-1"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "opacity-75 inline mr-2 " + icon.icon + " " + colors[icon.color]
  }), /*#__PURE__*/_react.default.createElement("p", {
    className: "inline text-lg"
  }, icon.text))), /*#__PURE__*/_react.default.createElement("div", {
    className: "py-6 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "shadow-lg mt-6 rounded-full my-6 mx-auto w-100-px p-6 bg-white"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: image,
    alt: "...",
    className: "mx-auto"
  })), /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-2xl font-bold leading-tight mt-0 mb-2"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-500 px-8"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center mt-8 mb-2 text-blueGray-400"
  }, /*#__PURE__*/_react.default.createElement(_ImagesOverlap.default, imagesOverlap)))));
}

CardProjectTeam.defaultProps = {
  icon: {},
  dropdown: {},
  imagesOverlap: {}
};
CardProjectTeam.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the ImagesOverlap component
  imagesOverlap: _propTypes.default.object,
  icon: _propTypes.default.shape({
    icon: _propTypes.default.string,
    text: _propTypes.default.string,
    color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
  })
};
