// @notus-pro/react/CardBilling.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardBilling;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames2 = _interopRequireDefault(require("classnames"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Select = _interopRequireDefault(require("./Select.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function CardBilling(_ref) {
  var title = _ref.title,
      inputs = _ref.inputs,
      paymentTitle = _ref.paymentTitle,
      defaultPaymentOpened = _ref.defaultPaymentOpened,
      paymentOptions = _ref.paymentOptions,
      onPaymentChange = _ref.onPaymentChange,
      paymentTabColor = _ref.paymentTabColor,
      returnButton = _ref.returnButton,
      orderButton = _ref.orderButton;

  var _React$useState = _react.default.useState(defaultPaymentOpened),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(false),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      oldInTransition = _React$useState4[0],
      setOldInTransition = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      newInTransition = _React$useState6[0],
      setNewInTransition = _React$useState6[1];

  var widths = {
    1: "lg:w-1/12",
    2: "lg:w-2/12",
    3: "lg:w-3/12",
    4: "lg:w-4/12",
    5: "lg:w-5/12",
    6: "lg:w-6/12",
    7: "lg:w-7/12",
    8: "lg:w-8/12",
    9: "lg:w-9/12",
    10: "lg:w-10/12",
    11: "lg:w-11/12",
    12: "lg:w-12/12"
  };
  var activeColors = {
    blueGray: "bg-blueGray-500 text-white",
    red: "bg-red-500 text-white",
    orange: "bg-orange-500 text-white",
    amber: "bg-amber-500 text-white",
    emerald: "bg-emerald-500 text-white",
    teal: "bg-teal-500 text-white",
    lightBlue: "bg-lightBlue-500 text-white",
    indigo: "bg-indigo-500 text-white",
    purple: "bg-purple-500 text-white",
    pink: "bg-pink-500 text-white"
  };
  var nonActiveColors = {
    blueGray: "text-blueGray-500 bg-white",
    red: "text-red-500 bg-white",
    orange: "text-orange-500 bg-white",
    amber: "text-amber-500 bg-white",
    emerald: "text-emerald-500 bg-white",
    teal: "text-teal-500 bg-white",
    lightBlue: "text-lightBlue-500 bg-white",
    indigo: "text-indigo-500 bg-white",
    purple: "text-purple-500 bg-white",
    pink: "text-pink-500 bg-white"
  };

  var toggleNew = function toggleNew(e, newOpen, text) {
    e.preventDefault();

    if (!newInTransition && !oldInTransition) {
      setOldInTransition(true);
      setTimeout(function () {
        setOpen(newOpen);
      }, 500);
      setTimeout(function () {
        setOldInTransition(false);
        setNewInTransition(true);
      }, 600);
      setTimeout(function () {
        setNewInTransition(false);
      }, 1100);
    }

    onPaymentChange(newOpen, text);
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("form", null, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-3xl font-semibold mt-4 mb-6"
  }, title), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, inputs.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "px-4 pb-2 relative w-full " + widths[prop.width]
    }, /*#__PURE__*/_react.default.createElement("label", {
      className: "block uppercase text-blueGray-700 text-xs font-bold mb-2 ml-1"
    }, prop.label), prop.input && /*#__PURE__*/_react.default.createElement(_Input.default, prop.input), prop.select && /*#__PURE__*/_react.default.createElement(_Select.default, prop.select));
  })), /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-3xl font-semibold mt-4 mb-6"
  }, paymentTitle), /*#__PURE__*/_react.default.createElement("ul", {
    className: "flex-col md:flex-row flex flex-wrap list-none pl-0 mb-0"
  }, paymentOptions.map(function (prop, key) {
    var _classnames;

    return /*#__PURE__*/_react.default.createElement("li", {
      key: key,
      className: "-mb-px mr-2 last:mr-0 flex-auto text-center"
    }, /*#__PURE__*/_react.default.createElement("a", {
      href: "#pablo",
      className: (0, _classnames2.default)("text-xs font-bold uppercase px-5 py-3 shadow rounded block leading-normal sm:mb-4 md:mb-0 uppercase duration-500 transition-all ease-in-out", (_classnames = {}, _defineProperty(_classnames, nonActiveColors[paymentTabColor], key !== open), _defineProperty(_classnames, activeColors[paymentTabColor], key === open), _classnames)),
      onClick: function onClick(e) {
        return toggleNew(e, key, prop.tabName);
      }
    }, prop.tabName));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full"
  }, paymentOptions.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: (0, _classnames2.default)("my-8 transform duration-500 transition-all ease-in-out", {
        hidden: key !== open,
        block: key === open,
        "opacity-0 scale-0": key === open && oldInTransition,
        "opacity-100 scale-100": key === open && newInTransition
      })
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex flex-wrap -mx-4"
    }, prop.inputs.map(function (inputProp, inputKey) {
      return /*#__PURE__*/_react.default.createElement("div", {
        key: inputKey,
        className: "px-4 relative w-full " + widths[inputProp.width]
      }, /*#__PURE__*/_react.default.createElement("label", {
        className: "block uppercase text-blueGray-700 text-xs font-bold mb-2 ml-1"
      }, inputProp.label), inputProp.input && /*#__PURE__*/_react.default.createElement(_Input.default, inputProp.input), inputProp.select && /*#__PURE__*/_react.default.createElement(_Select.default, inputProp.select));
    })));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-between mt-12 mb-8"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, returnButton), /*#__PURE__*/_react.default.createElement(_Button.default, orderButton)))))));
}

var inputsObject = {
  // NOTE: this width only applies on large devices
  width: _propTypes.default.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
  // if you wish, you can send somthing like
  // <span className="mr-2">Label Name</span><span className="text-red-500">*</span>
  // NOTE: the label tag will be auto generated
  label: _propTypes.default.node
};

var inputsShape = _propTypes.default.shape(_objectSpread(_objectSpread({}, inputsObject), {}, {
  // props to pass to the Input component
  input: _propTypes.default.object
}));

var selectsShape = _propTypes.default.shape(_objectSpread(_objectSpread({}, inputsObject), {}, {
  // props to pass to the Select component
  select: _propTypes.default.object
}));

var inputsTypes = _propTypes.default.arrayOf(_propTypes.default.oneOfType([inputsShape, selectsShape]));

CardBilling.defaultProps = {
  inputs: [],
  paymentOptions: [],
  onPaymentChange: function onPaymentChange() {},
  returnButton: {},
  orderButton: {},
  defaultPaymentOpened: 0,
  paymentTabColor: "blueGray"
};
CardBilling.propTypes = {
  title: _propTypes.default.string,
  inputs: inputsTypes,
  paymentTitle: _propTypes.default.string,
  // 0 represents the first element
  // also, you should note that
  // the number should not be lower then 0
  // or higher than the number of items - 1
  defaultPaymentOpened: _propTypes.default.number,
  paymentOptions: _propTypes.default.arrayOf(_propTypes.default.shape({
    tabName: _propTypes.default.string,
    inputs: inputsTypes
  })),
  // this function has two parameters,
  // index number of the tab that was pressed
  // and the tabName of that tab
  // example: onPaymentChange={(index,name) => console.log(index,name)}
  onPaymentChange: _propTypes.default.func,
  paymentTabColor: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  // props to pass to the Button component on the left
  returnButton: _propTypes.default.object,
  // props to pass to the Button component on the right
  orderButton: _propTypes.default.object
};
