// @notus-pro/react/CardProfileFull.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProfileFull;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function CardProfileFull(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      list = _ref.list,
      dropdown = _ref.dropdown;
  var colorClass = {
    blueGray: "text-blueGray-500 bg-blueGray-200",
    red: "text-red-500 bg-red-200",
    orange: "text-orange-500 bg-orange-200",
    amber: "text-amber-500 bg-amber-200",
    emerald: "text-emerald-500 bg-emerald-200",
    teal: "text-teal-500 bg-teal-200",
    lightBlue: "text-lightBlue-500 bg-lightBlue-200",
    indigo: "text-indigo-500 bg-indigo-200",
    purple: "text-purple-500 bg-purple-200",
    pink: "text-pink-500 bg-pink-200"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "shadow-lg rounded-lg relative flex flex-col min-w-0 break-words bg-white w-full mb-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    className: "w-full flex-shrink-0 rounded-t-lg",
    src: image
  }), /*#__PURE__*/_react.default.createElement("svg", {
    preserveAspectRatio: "none",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 583 95",
    className: "block w-full absolute text-white svg-bg",
    style: {
      height: "95px",
      bottom: "0"
    }
  }, /*#__PURE__*/_react.default.createElement("polygon", {
    points: "583,55 583,95 0,95",
    style: {
      fill: "currentColor"
    }
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xl font-bold"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-sm mt-1 mb-5 text-blueGray-400 font-bold uppercase"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "block w-full",
    style: {
      overflowX: "auto"
    }
  }, /*#__PURE__*/_react.default.createElement("ul", {
    className: "list-none"
  }, list.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("li", {
      className: "py-1",
      key: key
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex items-center"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "inline-flex text-center rounded-full mr-3 items-center justify-center w-8 h-8 text-sm " + colorClass[prop.color]
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: prop.icon
    })), /*#__PURE__*/_react.default.createElement("h6", {
      className: "text-md text-blueGray-500"
    }, prop.text)));
  }))))));
}

CardProfileFull.defaultProps = {
  list: [],
  dropdown: {}
};
CardProfileFull.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  list: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.string,
    text: _propTypes.default.string,
    color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
  })),
  // props that can be passed to the DropdownButton component
  dropdown: _propTypes.default.object
};
