// @notus-pro/react/CardProjectImageFullBg.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProjectImageFullBg;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardProjectImageFullBg(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      icon = _ref.icon,
      color = _ref.color,
      button = _ref.button,
      alignment = _ref.alignment;
  var colors = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  var alignmentsText = {
    left: "text-left",
    center: "text-center",
    right: "text-right"
  };
  var alignmentsIcon = {
    left: "mr-auto",
    center: "mx-auto",
    right: "ml-auto"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg bg-50 bg-cover p-10",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full top-0 left-0 bg-black opacity-60 z-0 rounded-lg"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: alignmentsText[alignment] + " flex-auto z-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: alignmentsIcon[alignment] + " mb-6 bg-white rounded-full w-16 h-16 text-center inline-flex items-center justify-center "
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: colors[color] + " text-2xl " + icon
  })), /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-3xl font-bold mt-0 mb-2 text-white leading-tight"
  }, title), /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-white opacity-75 text-xl mt-0 mb-4"
  }, description), /*#__PURE__*/_react.default.createElement(_Button.default, button))));
}

CardProjectImageFullBg.defaultProps = {
  button: {},
  color: "blueGray",
  alignment: "center"
};
CardProjectImageFullBg.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object,
  icon: _propTypes.default.string,
  alignment: _propTypes.default.oneOf(["left", "center", "right"]),
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
