// @notus-pro/react/CardProductPrice.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProductPrice;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardProductPrice(_ref) {
  var image = _ref.image,
      subtitle = _ref.subtitle,
      title = _ref.title,
      description = _ref.description,
      price = _ref.price,
      button = _ref.button;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-6"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "w-full"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-6 py-6 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-sm font-bold uppercase mt-0 mb-1 text-pink-500"
  }, subtitle), /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-2xl font-semibold mt-0 mb-2"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-500 leading-relaxed"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-between mt-2"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "text-blueGray-700 text-xl mt-1"
  }, price), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, button, {
    size: "sm"
  }))))));
}

CardProductPrice.defaultProps = {
  button: {}
};
CardProductPrice.propTypes = {
  image: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  price: _propTypes.default.string,
  // props to send to the Button component
  button: _propTypes.default.object
};
