// @notus-pro/react/CardBlogFullBgAuthor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardBlogFullBgAuthor;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardBlogFullBgAuthor(_ref) {
  var bgImage = _ref.bgImage,
      title = _ref.title,
      description = _ref.description,
      userimage = _ref.userimage,
      username = _ref.username,
      user = _ref.user,
      button = _ref.button;

  var userComponent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("img", {
    src: userimage,
    className: "rounded-full mr-4",
    style: {
      maxWidth: "40px"
    },
    alt: "..."
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between items-center"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-sm uppercase mb-2 mt-2 text-white font-semibold"
  }, username))));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "overflow-hidden mb-12 rounded-lg relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded h-550-px transition-all duration-150 ease-in-out hover:transform hover:scale-110 group"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute rounded-lg w-full h-full bg-50-center bg-cover transition-all duration-1000 ease-in-out group-hover:transform group-hover:scale-110",
    style: {
      backgroundImage: "url(" + require("assets/img/sections/unsplashs.jpg").default + ")",
      backfaceVisibility: "hidden"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute rounded-lg w-full h-full bg-black opacity-50"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "p-10 flex h-full items-end z-1"
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-4xl font-semibold mt-0 mb-2 text-white"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-white opacity-75 text-lg"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "py-4 mb-3 flex items-start"
  }, user.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, user, {
    className: "flex items-center"
  }), userComponent) : /*#__PURE__*/_react.default.createElement("a", _extends({}, user, {
    className: "flex items-center"
  }), userComponent)), /*#__PURE__*/_react.default.createElement(_Button.default, button)))));
}

CardBlogFullBgAuthor.defaultProps = {
  button: {},
  user: {}
};
CardBlogFullBgAuthor.propTypes = {
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  bgImage: _propTypes.default.string,
  userimage: _propTypes.default.string,
  username: _propTypes.default.string,
  // array of properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  // this will wrapp the username and userimage
  user: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the Button,
  // so please check that one out
  button: _propTypes.default.object
};
