// @notus-pro/react/CardBlogImageHorizontal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardBlogImageHorizontal;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

// components
function CardBlogImageHorizontal(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      link = _ref.link,
      user = _ref.user,
      username = _ref.username,
      userlink = _ref.userlink,
      date = _ref.date;

  var userComponent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("img", {
    src: user,
    className: "rounded-full mr-4 w-12 h-12",
    alt: "..."
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between items-center"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-md font-bold uppercase mt-0 mb-0 text-blueGray-500"
  }, username), /*#__PURE__*/_react.default.createElement("small", {
    className: "text-sm text-blueGray-500 mt-0"
  }, date))));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12 relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-4/12"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: image,
    className: "w-full rounded-lg rounded-r-none",
    alt: "..."
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-8/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "lg:pl-0 px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-3xl font-semibold text-blueGray-700 leading-tight"
  }, link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, {
    to: link.to
  }, title) : /*#__PURE__*/_react.default.createElement("a", {
    href: link.href
  }, title)), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-3 text-lg leading-relaxed text-blueGray-500"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "mr-2"
  }, description), link && link.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, link, {
    className: "text-lightBlue-500"
  })) : /*#__PURE__*/_react.default.createElement("a", _extends({}, link, {
    className: "text-lightBlue-500"
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-6"
  }, userlink && userlink.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, link, {
    className: "inline-flex items-center"
  }), userComponent) : /*#__PURE__*/_react.default.createElement("a", _extends({}, link, {
    className: "inline-flex items-center"
  }), userComponent)))))));
}

CardBlogImageHorizontal.defaultProps = {
  link: {},
  userlink: {}
};
CardBlogImageHorizontal.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // properties to pass to the link object after the description
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  // NOTE: this will also apply as a wrapper for the title
  link: _propTypes.default.object,
  // this is an image for the user
  user: _propTypes.default.string,
  username: _propTypes.default.string,
  // properties to pass to the link object that
  // wrapps the user, date and username
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  userlink: _propTypes.default.object,
  date: _propTypes.default.string
};
