// @notus-pro/react/CardInvoice.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardInvoice;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function CardInvoice(_ref) {
  var image = _ref.image,
      address = _ref.address,
      phone = _ref.phone,
      invoice = _ref.invoice,
      billing = _ref.billing,
      invoiceDate = _ref.invoiceDate,
      dueDate = _ref.dueDate,
      footerTitle = _ref.footerTitle,
      footerDescriprion = _ref.footerDescriprion,
      contact = _ref.contact,
      totalCount = _ref.totalCount,
      tableHead = _ref.tableHead,
      tableBody = _ref.tableBody;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-white shadow-lg rounded-lg relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-5 pt-6 pb-4 border-b border-blueGray-200"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left px-4 relative w-full md:w-4/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "w-full mb-2",
    style: {
      maxWidth: "150px"
    }
  }), /*#__PURE__*/_react.default.createElement("h6", {
    className: "block capitalize text-blueGray-700 mt-6"
  }, address), /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-blueGray-700"
  }, phone))), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left px-4 relative w-full lg:w-3/12 w-full md:w-5/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center py-6 lg:pt-6 pt-12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xl font-bold leading-normal mt-0 mb-2"
  }, billing.title), /*#__PURE__*/_react.default.createElement("h6", {
    className: "block mt-2 mb-0 text-xl"
  }, billing.name), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-500"
  }, billing.address))))), /*#__PURE__*/_react.default.createElement("div", {
    className: "md:justify-between flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left px-4 relative w-full md:w-4/12"
  }, /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-2xl font-semibold leading-normal mb-2 mt-12 text-left"
  }, invoice.text), /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-2xl font-normal leading-normal mt-1 mb-2 font-light"
  }, invoice.id)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-left px-4 relative w-full lg:w-3/12 w-full md:w-5/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center py-6 lg:pt-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-12"
  }, /*#__PURE__*/_react.default.createElement("p", {
    className: "float-left mb-0"
  }, invoiceDate.text), /*#__PURE__*/_react.default.createElement("p", {
    className: "ml-4 float-right mb-0"
  }, invoiceDate.date), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("p", {
    className: "float-left"
  }, dueDate.text), /*#__PURE__*/_react.default.createElement("p", {
    className: "ml-4 float-right"
  }, dueDate.date)))))), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, /*#__PURE__*/_react.default.createElement("table", {
    className: "text-right w-full mb-4 text-blueGray-800 border-collapse"
  }, /*#__PURE__*/_react.default.createElement("thead", {
    className: "bg-blueGray-800"
  }, /*#__PURE__*/_react.default.createElement("tr", {
    className: "text-right text-white uppercase font-light"
  }, tableHead.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("th", {
      key: key,
      className: "p-3 border-t",
      scope: "col"
    }, prop);
  }))), /*#__PURE__*/_react.default.createElement("tbody", null, tableBody.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      key: key
    }, prop.map(function (col, colKey) {
      return /*#__PURE__*/_react.default.createElement("td", {
        key: colKey,
        className: "py-4 p-3 border-t"
      }, col);
    }));
  })), /*#__PURE__*/_react.default.createElement("tfoot", null, /*#__PURE__*/_react.default.createElement("tr", {
    className: "mt-4"
  }, /*#__PURE__*/_react.default.createElement("th", {
    className: "border-b-0 p-3 border-t"
  }, /*#__PURE__*/_react.default.createElement("p", {
    className: "text-lg font-semibold pt-2"
  }, totalCount.text)), /*#__PURE__*/_react.default.createElement("th", {
    className: "border-b-0 p-3 border-t",
    colSpan: tableHead.length - 1 < 1 ? 1 : tableHead.length - 1
  }, /*#__PURE__*/_react.default.createElement("p", {
    className: "text-right text-lg font-semibold pt-2"
  }, totalCount.info)))))), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-3 border-t border-blueGray-200"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-right ml-auto px-4 relative w-full md:w-5/12"
  }, /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-2xl font-semibold leading-normal mt-0 mb-2"
  }, footerTitle), /*#__PURE__*/_react.default.createElement("p", {
    className: ""
  }, footerDescriprion), contact.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("h6", {
      key: key,
      className: "flex items-center justify-end mb-4"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "mr-2"
    }, prop.type), /*#__PURE__*/_react.default.createElement("span", {
      className: "text-blueGray-500"
    }, prop.info));
  })))));
}

CardInvoice.defaultProps = {
  invoice: {},
  billing: {},
  invoiceDate: {},
  dueDate: {},
  contact: [],
  tableHead: [],
  tableBody: [],
  totalCount: {}
};
CardInvoice.propTypes = {
  image: _propTypes.default.string,
  address: _propTypes.default.string,
  phone: _propTypes.default.string,
  invoice: _propTypes.default.shape({
    text: _propTypes.default.string,
    id: _propTypes.default.string
  }),
  billing: _propTypes.default.shape({
    title: _propTypes.default.string,
    name: _propTypes.default.string,
    address: _propTypes.default.string
  }),
  invoiceDate: _propTypes.default.shape({
    text: _propTypes.default.string,
    date: _propTypes.default.string
  }),
  dueDate: _propTypes.default.shape({
    text: _propTypes.default.string,
    date: _propTypes.default.string
  }),
  footerTitle: _propTypes.default.string,
  footerDescriprion: _propTypes.default.string,
  contact: _propTypes.default.arrayOf(_propTypes.default.shape({
    type: _propTypes.default.string,
    info: _propTypes.default.string
  })),
  totalCount: _propTypes.default.shape({
    text: _propTypes.default.string,
    info: _propTypes.default.string
  }),
  tableHead: _propTypes.default.arrayOf(_propTypes.default.string),
  // example: [["one","two"],["three","four","five"]]
  // will be generated as a table with two rows,
  // the first row with two columns -         one | two
  // the second row with three columns - three | four | five
  tableBody: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.string))
};
