// @notus-pro/react/CardLoginSocial.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardLoginSocial;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _Button = _interopRequireDefault(require("./Button.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Checkbox = _interopRequireDefault(require("./Checkbox.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CardLoginSocial(_ref) {
  var title = _ref.title,
      subtitle = _ref.subtitle,
      inputs = _ref.inputs,
      socials = _ref.socials,
      button = _ref.button,
      checkbox = _ref.checkbox,
      forgotPassword = _ref.forgotPassword,
      createAccount = _ref.createAccount;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col w-full mb-6 shadow-lg rounded-lg bg-white"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-0 px-6 py-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mb-3"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-blueGray-500 text-sm font-bold"
  }, title)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center"
  }, socials.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      size: "sm"
    }, prop.button, {
      key: key,
      color: prop.icon,
      fullWidth: false
    }), /*#__PURE__*/_react.default.createElement("i", {
      className: "mr-1 fab fa-" + prop.icon
    }), " ", prop.icon);
  })), /*#__PURE__*/_react.default.createElement("hr", {
    className: "mt-6 border-b-1 border-blueGray-200"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-auto px-4 lg:px-10 py-10 pt-0"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-blueGray-500 text-center mb-3 font-bold"
  }, /*#__PURE__*/_react.default.createElement("small", null, subtitle)), /*#__PURE__*/_react.default.createElement("form", null, inputs.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "relative w-full"
    }, /*#__PURE__*/_react.default.createElement("label", {
      className: "block uppercase text-blueGray-500 text-xs font-bold mb-2 ml-1"
    }, prop.label), /*#__PURE__*/_react.default.createElement(_Input.default, prop.input));
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-2 inline-block"
  }, /*#__PURE__*/_react.default.createElement(_Checkbox.default, checkbox)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mt-5"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button))))), Object.keys(forgotPassword).length === 0 && Object.keys(createAccount).length === 0 ? null : /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap mt-6 relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-1/2"
  }, forgotPassword && forgotPassword.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, forgotPassword, {
    className: "text-blueGray-500 ml-2"
  }), /*#__PURE__*/_react.default.createElement("small", null, "Forgot password?")) : forgotPassword && forgotPassword.href ? /*#__PURE__*/_react.default.createElement("a", _extends({}, forgotPassword, {
    className: "text-blueGray-500 ml-2"
  }), /*#__PURE__*/_react.default.createElement("small", null, "Forgot password?")) : null), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-1/2 text-right"
  }, createAccount && createAccount.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, createAccount, {
    className: "text-blueGray-500 mr-2"
  }), /*#__PURE__*/_react.default.createElement("small", null, "Create new account")) : createAccount && createAccount.href ? /*#__PURE__*/_react.default.createElement("a", _extends({}, createAccount, {
    className: "text-blueGray-500 mr-2"
  }), /*#__PURE__*/_react.default.createElement("small", null, "Create new account")) : null)));
}

CardLoginSocial.defaultProps = {
  inputs: [],
  button: {},
  forgotPassword: {},
  createAccount: {}
};
CardLoginSocial.propTypes = {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  // NOTE: the "Forgot password?" text is allready set
  forgotPassword: _propTypes.default.object,
  // NOTE: the "Create new account" text is allready set
  createAccount: _propTypes.default.object,
  // It is represetnted by the props you
  // can pass to our Button component element
  button: _propTypes.default.object,
  socials: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.oneOf(["facebook", "twitter", "instagram", "github", "pinterest", "youtube", "vimeo", "slack", "dribbble", "reddit", "tumblr", "linkedin"]),
    // It is represetnted by the props you
    // can pass to our Button component element
    // for example you can add an onClick handler
    button: _propTypes.default.object
  })),
  // It is represetnted by the props you
  // can pass to our Checkbox component element
  checkbox: _propTypes.default.object,
  inputs: _propTypes.default.arrayOf(_propTypes.default.shape({
    label: _propTypes.default.string,
    // It is represetnted by the props you
    // can pass to our Input component element
    // NOTE: if you wish to generate a textarea, you will need to pass
    // // // inside this object >> type: "textarea"
    // // // full example >> input: { type: "textarea" }
    input: _propTypes.default.object
  }))
};
