// @notus-pro/react/CardProductsCheckout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = CardProductsCheckout;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function CardProductsCheckout(_ref) {
  var images = _ref.images,
      discount = _ref.discount,
      details = _ref.details;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex flex-col min-w-0 break-words bg-white w-full mb-6 shadow-lg rounded-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 py-5 flex-auto"
  }, images.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "items-center my-4 flex items-start"
    }, /*#__PURE__*/_react.default.createElement("img", {
      src: prop.image,
      className: "mr-4 w-20 p-2",
      alt: "..."
    }), /*#__PURE__*/_react.default.createElement("div", {
      className: "flex-1"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "justify-between"
    }, /*#__PURE__*/_react.default.createElement("h6", {
      className: "text-xl font-semibold leading-tight mb-1"
    }, prop.title), /*#__PURE__*/_react.default.createElement("p", {
      className: "text-blueGray-500 uppercase font-bold text-xs"
    }, prop.subtitle))), /*#__PURE__*/_react.default.createElement("span", {
      className: "text-blueGray-700"
    }, prop.price));
  }), discount && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("hr", {
    className: "mt-6 mb-4 bg-blueGray-300 ml-0"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-0 pt-0 relative"
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "block uppercase text-blueGray-700 text-xs font-bold mb-2 ml-1"
  }, discount.text), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-2 relative flex flex-wrap w-full items-stretch"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mr-2"
  }, /*#__PURE__*/_react.default.createElement(_Input.default, discount.input)), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-0"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, discount.button))))), details.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key
    }, /*#__PURE__*/_react.default.createElement("hr", {
      className: "mt-4 mb-2 bg-blueGray-300 ml-0"
    }), prop.map(function (propDetails, propKey) {
      return /*#__PURE__*/_react.default.createElement("div", {
        key: propKey,
        className: "flex justify-between"
      }, /*#__PURE__*/_react.default.createElement("h6", {
        className: "text-blueGray-700 leading-normal mt-0 mb-2"
      }, propDetails.text), /*#__PURE__*/_react.default.createElement("h6", {
        className: "leading-normal mt-0 mb-2"
      }, propDetails.info));
    }));
  }))));
}

CardProductsCheckout.defaultProps = {
  images: [],
  details: []
};
CardProductsCheckout.propTypes = {
  images: _propTypes.default.arrayOf(_propTypes.default.shape({
    image: _propTypes.default.string,
    title: _propTypes.default.string,
    subtitle: _propTypes.default.string,
    price: _propTypes.default.string
  })),
  discount: _propTypes.default.shape({
    text: _propTypes.default.string,
    // props to pass to the Input component
    input: _propTypes.default.object,
    // props to pass to the Button component
    button: _propTypes.default.object
  }),
  details: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.shape({
    text: _propTypes.default.string,
    info: _propTypes.default.string
  })))
};
