// @notus-pro/react/Comment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Comment;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Comment(_ref) {
  var image = _ref.image,
      username = _ref.username,
      description = _ref.description,
      likes = _ref.likes,
      shares = _ref.shares;

  var likesComponent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("i", {
    className: "fas fa-thumbs-up"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "text-blueGray-500"
  }, " ", likes.children, " likes "));

  var sharesComponent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("i", {
    className: "fas fa-share"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "text-blueGray-500"
  }, " ", shares.children, " shares "));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-2 flex items-start group border-b border-blueGray-200"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "relative rounded-full w-10 h-10 mt-3"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1 relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-5"
  }, /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-lg font-bold mt-0 mb-2"
  }, username), /*#__PURE__*/_react.default.createElement("p", {
    className: "mb-4 text-blueGray-500"
  }, description), /*#__PURE__*/_react.default.createElement("div", null, likes.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, likes, {
    className: "inline-block mr-3 text-blueGray-400 font-semibold text-sm"
  }), likesComponent) : /*#__PURE__*/_react.default.createElement("a", _extends({}, likes, {
    className: "inline-block mr-3 text-blueGray-400 font-semibold text-sm"
  }), likesComponent), shares.to ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, shares, {
    className: "inline-block mr-3 text-blueGray-400 font-semibold text-sm"
  }), sharesComponent) : /*#__PURE__*/_react.default.createElement("a", _extends({}, shares, {
    className: "inline-block mr-3 text-blueGray-400 font-semibold text-sm"
  }), sharesComponent))))));
}

Comment.defaultProps = {
  likes: {
    children: "0"
  },
  shares: {
    children: "0"
  }
};
Comment.propTypes = {
  username: _propTypes.default.string,
  description: _propTypes.default.string,
  image: _propTypes.default.string,
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  // NOTE: the like icon is set by default,
  // // // you only need to set the number of like
  likes: _propTypes.default.object,
  // properties to pass to the link object
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  // NOTE: the share icon is set by default,
  // // // you only need to set the number of comments
  shares: _propTypes.default.object
};
