// @notus-pro/react/Contact.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Contact;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Contact(_ref) {
  var image = _ref.image,
      name = _ref.name,
      color = _ref.color,
      lastOnline = _ref.lastOnline,
      lastMessage = _ref.lastMessage,
      link = _ref.link;
  var bgColors = {
    light: "bg-blueGray-200 text-blueGray-800",
    dark: "bg-blueGray-800 text-white",
    blueGray: "bg-blueGray-500 text-white",
    red: "bg-red-500 text-white",
    orange: "bg-orange-500 text-white",
    amber: "bg-amber-500 text-white",
    emerald: "bg-emerald-500 text-white",
    teal: "bg-teal-500 text-white",
    lightBlue: "bg-lightBlue-500 text-white",
    indigo: "bg-indigo-500 text-white",
    purple: "bg-purple-500 text-white",
    pink: "bg-pink-500 text-white"
  };
  var lastOnlineColors = {
    light: "text-blueGray-500",
    dark: "text-blueGray-200",
    blueGray: "text-blueGray-200",
    red: "text-blueGray-200",
    orange: "text-blueGray-200",
    amber: "text-blueGray-200",
    emerald: "text-blueGray-200",
    teal: "text-blueGray-200",
    lightBlue: "text-blueGray-200",
    indigo: "text-blueGray-200",
    purple: "text-blueGray-200",
    pink: "text-blueGray-200"
  };
  var lastMessageColors = {
    light: "text-blueGray-500",
    dark: "text-blueGray-100",
    blueGray: "text-blueGray-100",
    red: "text-blueGray-100",
    orange: "text-blueGray-100",
    amber: "text-blueGray-100",
    emerald: "text-blueGray-100",
    teal: "text-blueGray-100",
    lightBlue: "text-blueGray-100",
    indigo: "text-blueGray-100",
    purple: "text-blueGray-100",
    pink: "text-blueGray-100"
  };

  var details = /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center p-3"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: image,
    className: "rounded-full max-w-40-px",
    alt: "..."
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between items-center"
  }, /*#__PURE__*/_react.default.createElement("h6", {
    className: "text-lg font-semibold leading-normal pl-3 inline-block"
  }, name), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-200 uppercase font-bold text-xs inline-block float-right " + lastOnlineColors[color]
  }, lastOnline, lastOnline === "Online" && /*#__PURE__*/_react.default.createElement("span", {
    className: "p-1 mt-2 ml-2 text-xs font-semibold inline-block uppercase last:mr-0 mr-1 p-1 rounded bg-emerald-400"
  }), lastOnline === "Offline" && /*#__PURE__*/_react.default.createElement("span", {
    className: "p-1 mt-2 ml-2 text-xs font-semibold inline-block uppercase last:mr-0 mr-1 p-1 rounded bg-red-500"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "pl-3"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "block truncate " + lastMessageColors[color]
  }, lastMessage)))));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-4 rounded-md relative block my-1 " + bgColors[color]
  }, link && link.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, link, details), link && link.to === undefined && /*#__PURE__*/_react.default.createElement("a", link, details)));
}

Contact.defaultProps = {};
Contact.propTypes = {
  image: _propTypes.default.string,
  name: _propTypes.default.string,
  color: _propTypes.default.oneOf(["light", "dark", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  // if Online, then a emerald bullet will appear near it
  // if Offline, then a red bullet will appear near it
  lastOnline: _propTypes.default.string,
  lastMessage: _propTypes.default.string,
  // properties to pass to the link object
  // that wrapps the whole component
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object
};
