// @notus-pro/react/ContactUs2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ContactUs2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Heading = _interopRequireDefault(require("./Heading2.js"));

var _InfoAreaIcon = _interopRequireDefault(require("./InfoAreaIcon.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function ContactUs2(_ref) {
  var image = _ref.image,
      heading2 = _ref.heading2,
      button = _ref.button,
      infoAreas = _ref.infoAreas;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "overflow-hidden contact-us-3 py-20"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-0 flex items-center relative min-h-400-px h-400-px max-h-999-px mb-50-px"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    className: "w-full",
    src: image
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "pt-12 container mx-auto px-4"
  }, heading2 && /*#__PURE__*/_react.default.createElement(_Heading.default, heading2), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4 items-center justify-center -mt-6"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button)), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4 mt-12"
  }, infoAreas.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: (0, _classnames.default)("px-4 relative w-full lg:w-3/12 w-full md:w-6/12 w-6/12", {
        "border-r": key !== infoAreas.length - 1
      })
    }, /*#__PURE__*/_react.default.createElement(_InfoAreaIcon.default, prop));
  })))));
}

ContactUs2.defaultProps = {
  button: {},
  infoAreas: [],
  heading2: null
};
ContactUs2.propTypes = {
  image: _propTypes.default.string,
  // prop to pass to the Heading2 component
  heading2: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the Button,
  // so please check that one out
  button: _propTypes.default.object,
  // it is an array that has objects which
  // are represented by the props
  // that can be passed to the InfoAreaIcon,
  // so please check that one out
  infoAreas: _propTypes.default.arrayOf(_propTypes.default.object)
};
