// @notus-pro/react/ContactUs4.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ContactUs4;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading2Badge = _interopRequireDefault(require("./Heading2Badge.js"));

var _CardContactUsAddress = _interopRequireDefault(require("./CardContactUsAddress.js"));

var _InfoAreaIconTitle = _interopRequireDefault(require("./InfoAreaIconTitle.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ContactUs4(_ref) {
  var heading2badge = _ref.heading2badge,
      image = _ref.image,
      infoareas = _ref.infoareas,
      card = _ref.card;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-900 py-20",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto flex justify-center flex-col text-white px-4 relative lg:w-4/12 w-full md:w-5/12"
  }, heading2badge && /*#__PURE__*/_react.default.createElement(_Heading2Badge.default, heading2badge), infoareas.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_InfoAreaIconTitle.default, _extends({
      key: key
    }, prop));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto px-4 relative w-full lg:w-5/12 w-full md:w-7/12"
  }, /*#__PURE__*/_react.default.createElement(_CardContactUsAddress.default, card))))));
}

ContactUs4.defaultProps = {
  card: {},
  infoareas: [],
  heading2badge: null
};
ContactUs4.propTypes = {
  // prop to pass to the Heading2Badge component
  heading2badge: _propTypes.default.object,
  image: _propTypes.default.string,
  // array of props for the InfoAreaIconTitle component
  infoareas: _propTypes.default.arrayOf(_propTypes.default.object),
  // array of props for the CardContactUsAddress component
  card: _propTypes.default.object
};
