// @notus-pro/react/ContactUs3.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ContactUs3;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading2Badge = _interopRequireDefault(require("./Heading2Badge.js"));

var _CardContactUsIcons = _interopRequireDefault(require("./CardContactUsIcons.js"));

var _Map = _interopRequireDefault(require("./Map.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function ContactUs3(_ref) {
  var heading2badge = _ref.heading2badge,
      card = _ref.card,
      map = _ref.map,
      skew = _ref.skew;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-100"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full px-16 mx-auto px-4 mt-12 mb-6 pt-16"
  }, heading2badge && /*#__PURE__*/_react.default.createElement(_Heading2Badge.default, heading2badge)), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardContactUsIcons.default, card)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative overflow-hidden -mt-220-px"
  }, /*#__PURE__*/_react.default.createElement(_Map.default, map), /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full top-0"
  })));
}

ContactUs3.defaultProps = {
  heading2badge: null,
  infoAreas: [],
  map: {}
};
ContactUs3.propTypes = {
  // prop to pass to the Heading2Badge component
  heading2badge: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the Button,
  // so please check that one out
  badge: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the CardContactUsIcons,
  // so please check that one out
  card: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the Map,
  // so please check that one out
  map: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the Skew,
  // so please check that one out
  skew: _propTypes.default.object
};
