// @notus-pro/react/Error1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Error1;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Error1(_ref) {
  var image = _ref.image,
      color = _ref.color,
      error = _ref.error,
      description = _ref.description,
      title = _ref.title;
  var colors = {
    light: "text-blueGray-100",
    dark: "text-blueGray-800",
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  var className = "font-bold text-10 sm:text-12 tracking-875 " + colors[color];
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "h-screen overflow-hidden"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute bg-cover bg-50 h-full w-full",
    style: {
      backgroundImage: "url(" + image + ")"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 h-full"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4 h-full items-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center px-4 relative w-full"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: className
  }, error), /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-3xl font-bold mb-2 mt-0"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-2xl font-normal leading-normal text-blueGray-500"
  }, description))))));
}

Error1.defaulProps = {
  color: "dark"
};
Error1.propTypes = {
  error: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // background image to be added to the section
  image: _propTypes.default.string,
  // this is the colors for the error name
  color: _propTypes.default.oneOf(["light", "dark", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
