// @notus-pro/react/Features2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Features2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading2Badge = _interopRequireDefault(require("./Heading2Badge.js"));

var _CardInfoIconLeft = _interopRequireDefault(require("./CardInfoIconLeft.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Features2(_ref) {
  var heading2badge = _ref.heading2badge,
      image = _ref.image,
      cards = _ref.cards;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative my-12 bg-blueGray-800 text-center py-20",
    style: {
      backgroundImage: "url(" + image + ")"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading2badge && /*#__PURE__*/_react.default.createElement(_Heading2Badge.default, heading2badge), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-12 flex flex-wrap -mx-4"
  }, cards.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "p-4 relative w-full lg:w-6/12"
    }, /*#__PURE__*/_react.default.createElement(_CardInfoIconLeft.default, prop));
  })))));
}

Features2.defaultProps = {
  cards: [],
  color: "blueGray",
  heading2badge: null
};
Features2.propTypes = {
  // prop to pass to the Heading2Badge component
  heading2badge: _propTypes.default.object,
  image: _propTypes.default.string,
  // array of props to pass to the CardInfoIconLeft component
  cards: _propTypes.default.arrayOf(_propTypes.default.object)
};
