// @notus-pro/react/Features4.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Features4;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading3Badge = _interopRequireDefault(require("./Heading3Badge.js"));

var _CardInfoIconLeft = _interopRequireDefault(require("./CardInfoIconLeft.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Features4(_ref) {
  var heading3badge = _ref.heading3badge,
      image = _ref.image,
      cards = _ref.cards;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative py-20"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading3badge && /*#__PURE__*/_react.default.createElement(_Heading3Badge.default, heading3badge), /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-6/12"
  }, cards.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_CardInfoIconLeft.default, _extends({
      key: key
    }, prop));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "md:mx-auto px-4 relative w-full lg:w-6/12 w-10/12"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "lg:ml-12 w-full h-auto"
  }))))));
}

Features4.defaultProps = {
  cards: [],
  heading3badge: null
};
Features4.propTypes = {
  // prop to pass to the Heading3Badge component
  heading3badge: _propTypes.default.object,
  image: _propTypes.default.string,
  // array of props to pass to the CardInfoIconLeft component
  cards: _propTypes.default.arrayOf(_propTypes.default.object)
};
