// @notus-pro/react/HeaderImageTitleSmallCenter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderImageTitleSmallCenter;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Skew = _interopRequireDefault(require("./Skew.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderImageTitleSmallCenter(_ref) {
  var title = _ref.title,
      subtitle = _ref.subtitle,
      image = _ref.image,
      skewColor = _ref.skewColor;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("section", {
    className: "flex relative items-center p-0 min-h-screen-60 max-h-440-px"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full block bg-blueGray-900 opacity-75 z-1 left-0 top-0"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-800 w-full h-full absolute bg-cover bg-50 z-0",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative h-full text-center text-white container mx-auto px-4 z-3 mb-16"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-12 px-4 relative w-full lg:w-6/12 w-full md:w-8/12"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-4xl font-bold leading-tight"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-lg opacity-75 pt-2"
  }, subtitle)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full bottom-0 absolute z-2"
  }, /*#__PURE__*/_react.default.createElement(_Skew.default, {
    color: skewColor
  }))));
}

HeaderImageTitleSmallCenter.defaultProps = {
  skewColor: "white"
};
HeaderImageTitleSmallCenter.propTypes = {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  image: _propTypes.default.string,
  skewColor: _propTypes.default.oneOf(["white", "blueGray", "dark"])
};
