// @notus-pro/react/HeaderImageTitleButtons.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderImageTitleButtons;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Skew = _interopRequireDefault(require("./Skew.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderImageTitleButtons(_ref) {
  var image = _ref.image,
      title = _ref.title,
      socials = _ref.socials,
      skew = _ref.skew;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative section min-h-screen-75 flex items-center bg-cover",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-black opacity-75"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center justify-center container mx-auto px-4 h-full"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center mx-auto px-4 relative w-full lg:w-7/12"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-4xl font-bold leading-tight mt-0 mb-6 text-white"
  }, title), socials.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, {
      key: key,
      color: "pink",
      size: "sm",
      href: prop.link,
      target: "_blank"
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: "fab fa-" + prop.icon
    }));
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full absolute text-white bottom-0 z-1"
  }, /*#__PURE__*/_react.default.createElement(_Skew.default, skew))));
}

HeaderImageTitleButtons.defaultProps = {
  socials: []
};
HeaderImageTitleButtons.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  // this will generate an anchor with target blank to the given link
  socials: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.oneOf(["facebook", "twitter", "instagram", "dribbble", "github"]),
    link: _propTypes.default.string
  }))
};
