// @notus-pro/react/HeaderStatCards.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderStatCards;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _CardStatsMini = _interopRequireDefault(require("./CardStatsMini.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderStatCards(_ref) {
  var cards = _ref.cards,
      color = _ref.color;
  var bgColors = {
    light: "bg-blueGray-200",
    dark: "bg-blueGray-800",
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative pt-32 pb-32 " + bgColors[color]
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 md:px-6 mx-auto w-full"
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap"
  }, cards.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "w-full lg:w-6/12 xl:w-3/12 px-4"
    }, /*#__PURE__*/_react.default.createElement(_CardStatsMini.default, prop));
  }))))));
}

HeaderStatCards.defaultProps = {
  cards: [],
  color: "blueGray"
};
HeaderStatCards.propTypes = {
  // array of props to pass to the CardStatsMini component
  cards: _propTypes.default.arrayOf(_propTypes.default.object),
  color: _propTypes.default.oneOf(["light", "dark", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
