// @notus-pro/react/HeaderBlogPost.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderBlogPost;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Skew = _interopRequireDefault(require("./Skew.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderBlogPost(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      icon = _ref.icon,
      color = _ref.color,
      skew = _ref.skew;
  var iconBgColor = {
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-0 flex items-center overflow-hidden relative min-h-screen-75"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-cover bg-50 z-2",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-blueGray-900 z-1"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 z-3 pt-20 relative w-6/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-center flex flex-wrap -mx-4 text-center"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-5xl font-bold text-white leading-tight mt-0 mb-24"
  }, title))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full absolute text-white bottom-0 z-3"
  }, /*#__PURE__*/_react.default.createElement(_Skew.default, skew))), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 z-3 -mt-64 pt-20 relative w-8/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-white rounded-lg shadow-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center -mt-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: iconBgColor[color] + " rounded-full h-16 w-16 inline-flex items-center justify-center"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-xl text-white " + icon
  }))), /*#__PURE__*/_react.default.createElement("p", {
    className: "pt-6 pb-12 px-12 text-lg leading-relaxed text-center"
  }, description)))));
}

HeaderBlogPost.defaultProps = {
  color: "blueGray",
  skew: {}
};
HeaderBlogPost.propTypes = {
  // background image source
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  icon: _propTypes.default.string,
  // the background color of the icon
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  // props to pass to the Skew component
  skew: _propTypes.default.object
};
