// @notus-pro/react/HeaderImage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderImage;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Skew = _interopRequireDefault(require("./Skew.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderImage(_ref) {
  var image = _ref.image,
      skew = _ref.skew;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("section", {
    className: "relative block h-500-px"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute top-0 w-full h-full bg-center bg-cover",
    style: {
      backgroundImage: "url('https://images.unsplash.com/photo-1499336315816-097655dcfbda?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2710&q=80')"
    }
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "w-full h-full absolute opacity-50 bg-black"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full absolute top-auto bottom-0"
  }, /*#__PURE__*/_react.default.createElement(_Skew.default, skew))));
}

HeaderImage.defaultProps = {
  skew: {}
};
HeaderImage.propTypes = {
  image: _propTypes.default.string,
  // object of props that can be passed to the Skew component
  skew: _propTypes.default.object
};
