// @notus-pro/react/HeaderUser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderUser;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderUser(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      button = _ref.button,
      color = _ref.color,
      gradient = _ref.gradient;
  var bgColors = {
    blueGray: "bg-blueGray-800",
    red: "bg-red-800",
    orange: "bg-orange-800",
    amber: "bg-amber-800",
    emerald: "bg-emerald-800",
    teal: "bg-teal-800",
    lightBlue: "bg-lightBlue-800",
    indigo: "bg-indigo-800",
    purple: "bg-purple-800",
    pink: "bg-pink-800",
    "blueGray-gradient": "bg-gradient-to-r from-blueGray-800 to-blueGray-900",
    "red-gradient": "bg-gradient-to-r from-red-800 to-red-900",
    "orange-gradient": "bg-gradient-to-r from-orange-800 to-orange-900",
    "amber-gradient": "bg-gradient-to-r from-amber-800 to-amber-900",
    "emerald-gradient": "bg-gradient-to-r from-emerald-800 to-emerald-900",
    "teal-gradient": "bg-gradient-to-r from-teal-800 to-teal-900",
    "lightBlue-gradient": "bg-gradient-to-r from-lightBlue-800 to-lightBlue-900",
    "indigo-gradient": "bg-gradient-to-r from-indigo-800 to-indigo-900",
    "purple-gradient": "bg-gradient-to-r from-purple-800 to-purple-900",
    "pink-gradient": "bg-gradient-to-r from-pink-800 to-pink-900"
  };
  var gradientText = gradient ? "-gradient" : "";
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "min-h-500-px bg-cover bg-center-top relative pb-20 items-center flex text-white",
    style: {
      backgroundImage: "url(" + image + ")"
    }
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "opacity-90 absolute top-0 left-o w-full h-full " + bgColors[color + gradientText]
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "md:px-8 px-4 items-center flex w-full mx-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full px-4 md:w-10/12 lg:w-7/12"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "font-semibold leading-6 text-5xl mb-4 mt-0"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-0 mb-12"
  }, description), /*#__PURE__*/_react.default.createElement(_Button.default, button))))));
}

HeaderUser.defaultProps = {
  button: {},
  color: "blueGray",
  gradient: false
};
HeaderUser.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object,
  // whether the color should be a simple one, or a gradient one
  gradient: _propTypes.default.bool,
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
