// @notus-pro/react/HeaderNewsletterIllustration.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderNewsletterIllustration;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderNewsletterIllustration(_ref) {
  var image = _ref.image,
      button = _ref.button,
      input = _ref.input,
      title = _ref.title,
      description = _ref.description;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-0 flex items-center overflow-hidden min-h-screen-75"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center text-left flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-6/12"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-4xl font-bold leading-tight"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-4 mb-4 text-xl leading-relaxed text-blueGray-400"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4 mt-8"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 pr-2 relative w-full sm:w-8/12"
  }, /*#__PURE__*/_react.default.createElement(_Input.default, input)), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 pl-0 relative w-full sm:w-4/12"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-6/12"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: image,
    className: "w-full lg:ml-12",
    alt: "..."
  }))))));
}

HeaderNewsletterIllustration.defaultProps = {
  button: {},
  input: {}
};
HeaderNewsletterIllustration.propTypes = {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  description: _propTypes.default.string,
  image: _propTypes.default.string,
  // it is represented by the props
  // that can be passed to the Button,
  // so please check that one out
  button: _propTypes.default.object,
  // it is represented by the props
  // that can be passed to the Input,
  // so please check that one out
  input: _propTypes.default.object
};
