// @notus-pro/react/HeaderImageTitleButtonCenter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderImageTitleButtonCenter;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Skew = _interopRequireDefault(require("./Skew.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderImageTitleButtonCenter(_ref) {
  var image = _ref.image,
      title = _ref.title,
      icon = _ref.icon,
      color = _ref.color,
      text = _ref.text,
      skew = _ref.skew;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("section", {
    className: "flex relative items-center p-0 min-h-screen-75 z-0"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full block bg-black opacity-50 z-1 left-0 top-0"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-800 w-full h-full absolute bg-cover bg-50",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative h-full text-center text-white container mx-auto px-4 z-2 mb-24"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-12 px-4 relative w-full md:w-8/12"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-5xl font-bold leading-tight mt-0 mb-2"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-xl leading-relaxed opacity-75 mt-1 mb-4"
  }, text), /*#__PURE__*/_react.default.createElement(_Button.default, {
    color: color
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: icon + " mr-1"
  }), " Start shopping")))), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full bottom-0 absolute z-2"
  }, /*#__PURE__*/_react.default.createElement(_Skew.default, skew))));
}

HeaderImageTitleButtonCenter.defaultProps = {
  color: "orange",
  skew: {}
};
HeaderImageTitleButtonCenter.propTypes = {
  // background image source
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  text: _propTypes.default.string,
  icon: _propTypes.default.string,
  // props to pass to the Skew component
  skew: _propTypes.default.object,
  // the background color of the icon
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
