// @notus-pro/react/HeaderLeftTitleRightImagePlayer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderLeftTitleRightImagePlayer;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _InfoAreaIconTitle = _interopRequireDefault(require("./InfoAreaIconTitle.js"));

var _MediaPlayerImageOnly = _interopRequireDefault(require("./MediaPlayerImageOnly.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function HeaderLeftTitleRightImagePlayer(_ref) {
  var color = _ref.color,
      title = _ref.title,
      infoAreas = _ref.infoAreas,
      mediaplayer = _ref.mediaplayer;
  var colors = {
    blueGray: "bg-blueGray-800",
    red: "bg-red-800",
    orange: "bg-orange-800",
    amber: "bg-amber-800",
    emerald: "bg-emerald-800",
    teal: "bg-teal-800",
    lightBlue: "bg-lightBlue-800",
    indigo: "bg-indigo-800",
    purple: "bg-purple-800",
    pink: "bg-pink-800"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: colors[color]
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-0 flex items-center overflow-hidden relative min-h-screen max-h-999-px"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4 items-center"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement("h4", {
    className: "text-3xl font-bold mt-0 mb-12 pl-4 text-white"
  }, title), infoAreas.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_InfoAreaIconTitle.default, _extends({
      key: key
    }, prop));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_MediaPlayerImageOnly.default, mediaplayer)))))));
}

HeaderLeftTitleRightImagePlayer.defaultProps = {
  infoAreas: [],
  mediaplayer: {},
  color: "blueGray"
};
HeaderLeftTitleRightImagePlayer.propTypes = {
  title: _propTypes.default.string,
  // array of props that can be passed to the InfoAreaIconTitle component
  infoAreas: _propTypes.default.arrayOf(_propTypes.default.object),
  // props to pass to the MediaPlayerImageOnly component
  mediaplayer: _propTypes.default.object,
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
