// @notus-pro/react/Heading3Badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Heading3Badge;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Badge = _interopRequireDefault(require("./Badge.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Heading3Badge(_ref) {
  var badge = _ref.badge,
      title = _ref.title,
      description = _ref.description,
      color = _ref.color,
      alignment = _ref.alignment,
      whiteContent = _ref.whiteContent;
  var alignmentsRow = {
    left: "justify-start",
    center: "justify-center",
    right: "justify-end"
  };
  var alignmentsText = {
    left: "text-left",
    center: "text-center",
    right: "text-right"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12 flex flex-wrap -mx-4 " + alignmentsRow[alignment]
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-8/12 " + alignmentsText[alignment]
  }, badge && /*#__PURE__*/_react.default.createElement(_Badge.default, badge), title && /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-3xl font-bold mt-3 mb-1 " + (whiteContent ? "text-white" : "text-blueGray-700")
  }, title), description && /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-2 mb-4 text-xl leading-relaxed " + (whiteContent ? "text-white opacity-75" : "text-blueGray-500")
  }, description))));
}

Heading3Badge.defaultProps = {
  color: "blueGray",
  alignment: "left",
  whiteContent: false
};
Heading3Badge.propTypes = {
  // use this to make the title and description
  // change colors from blueGray-800 and blueGray-500
  // to white and blueGray-100
  whiteContent: _propTypes.default.bool,
  // props to pass to the Badge component
  badge: _propTypes.default.object,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  alignment: _propTypes.default.oneOf(["left", "center", "right"])
};
