// @notus-pro/react/HeaderLeftTitleRightImage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderLeftTitleRightImage;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function HeaderLeftTitleRightImage(_ref) {
  var image = _ref.image,
      title = _ref.title,
      description = _ref.description,
      button = _ref.button;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-0 flex items-center overflow-hidden min-h-screen-75"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute bg-contain bg-100 bg-no-repeat w-full h-full",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-5/12 w-full md:w-7/12",
    "ct-utillities": "mr-auto;text-left;d-flex;justify-content-center;flex-column"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-5xl font-bold leading-tight mt-0 mb-2"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-xl leading-relaxed opacity-75"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-8"
  }, /*#__PURE__*/_react.default.createElement(_Button.default, button)))))));
}

HeaderLeftTitleRightImage.defaultProps = {
  button: {}
};
HeaderLeftTitleRightImage.propTypes = {
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the Button component
  button: _propTypes.default.object
};
