// @notus-pro/react/HeaderBlogPostDark.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = HeaderBlogPostDark;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function HeaderBlogPostDark(_ref) {
  var image = _ref.image,
      title = _ref.title,
      date = _ref.date,
      quote = _ref.quote,
      quoteSource = _ref.quoteSource,
      description = _ref.description,
      icon = _ref.icon,
      color = _ref.color;
  var iconBgColor = {
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-0 flex items-center relative min-h-screen-75"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-cover bg-50 z-1",
    style: {
      backgroundImage: "url('" + image + "')"
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "absolute w-full h-full bg-black opacity-40 z-2"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 z-3"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-center text-white flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "text-4xl font-bold leading-tight mt-0 mb-2 mb-12"
  }, title)))), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 z-3 -mt-64 pt-20 relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-center text-white flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bg-blueGray-800 rounded-lg shadow-lg"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "text-center -mt-6"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: iconBgColor[color] + " rounded-full text-white w-16 h-16 inline-flex items-center justify-center -mt-2"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "p-1 text-xl " + icon
  }))), /*#__PURE__*/_react.default.createElement("blockquote", {
    className: "text-xl mx-0 mt-0 mb-4 block text-center mt-6"
  }, /*#__PURE__*/_react.default.createElement("small", {
    className: "font-light"
  }, date), /*#__PURE__*/_react.default.createElement("footer", {
    className: "text-base block text-blueGray-500"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "mr-2"
  }, quote), /*#__PURE__*/_react.default.createElement("cite", {
    title: "Source Title"
  }, quoteSource))), /*#__PURE__*/_react.default.createElement("p", {
    className: "mb-0 pt-6 pb-12 px-12 leading-relaxed text-center"
  }, description)))));
}

HeaderBlogPostDark.defaultProps = {
  color: "blueGray"
};
HeaderBlogPostDark.propTypes = {
  // background image source
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  date: _propTypes.default.string,
  quote: _propTypes.default.string,
  quoteSource: _propTypes.default.string,
  description: _propTypes.default.string,
  icon: _propTypes.default.string,
  // the background color of the icon
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
