// @notus-pro/react/Hero2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Hero2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading2Badge = _interopRequireDefault(require("./Heading2Badge.js"));

var _Skew = _interopRequireDefault(require("./Skew.js"));

var _InfoAreaIconTitleListColor = _interopRequireDefault(require("./InfoAreaIconTitleListColor.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Hero2(_ref) {
  var heading2badge = _ref.heading2badge,
      image = _ref.image,
      infoArea = _ref.infoArea;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("section", {
    className: "relative pb-20"
  }, /*#__PURE__*/_react.default.createElement(_Skew.default, null), /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4 pt-20 bg-white"
  }, /*#__PURE__*/_react.default.createElement(_Heading2Badge.default, heading2badge), /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex flex-wrap"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full md:w-4/12 ml-auto mr-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    className: "max-w-full rounded-lg shadow-lg",
    src: image
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full md:w-5/12 ml-auto mr-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "md:pr-12"
  }, /*#__PURE__*/_react.default.createElement(_InfoAreaIconTitleListColor.default, infoArea)))))));
}

Hero2.defaultProps = {
  heading2badge: null,
  infoArea: {}
};
Hero2.propTypes = {
  // prop to pass to the Heading2Badge component
  heading2badge: _propTypes.default.object,
  image: _propTypes.default.string,
  // it is represented by the props
  // that can be passed to the InfoAreaWhiteTitle,
  // so please check that one out
  infoAreas: _propTypes.default.arrayOf(_propTypes.default.object)
};
