// @notus-pro/react/Hero4.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Hero4;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading2Badge = _interopRequireDefault(require("./Heading2Badge.js"));

var _CardInfoImage = _interopRequireDefault(require("./CardInfoImage.js"));

var _CardInfoFullColor = _interopRequireDefault(require("./CardInfoFullColor.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Hero4(_ref) {
  var heading2badge = _ref.heading2badge,
      leftCardProps = _ref.leftCardProps,
      rightCardInfoIconProps = _ref.rightCardInfoIconProps,
      rightCardBlogImageProps = _ref.rightCardBlogImageProps,
      rightCardContactUsTitleProps = _ref.rightCardContactUsTitleProps,
      rightCardInfoImageProps = _ref.rightCardInfoImageProps;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-12"
  }, heading2badge && /*#__PURE__*/_react.default.createElement(_Heading2Badge.default, heading2badge), /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "ml-auto text-left mt-6 px-4 relative w-full lg:w-4/12"
  }, /*#__PURE__*/_react.default.createElement(_CardInfoImage.default, leftCardProps)), /*#__PURE__*/_react.default.createElement("div", {
    className: "mr-auto px-4 relative w-full lg:w-7/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardInfoFullColor.default, rightCardInfoIconProps)), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardInfoFullColor.default, rightCardBlogImageProps))), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardInfoFullColor.default, rightCardContactUsTitleProps)), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardInfoFullColor.default, rightCardInfoImageProps)))))));
}

Hero4.defaultProps = {
  heading2badge: null,
  leftCardProps: {},
  rightCardInfoIconProps: {},
  rightCardBlogImageProps: {},
  rightCardContactUsTitleProps: {},
  rightCardInfoImageProps: {}
};
Hero4.propTypes = {
  // prop to pass to the Heading2Badge component
  heading2badge: _propTypes.default.object,
  // Please check CardInfoImage props for more info
  leftCardProps: _propTypes.default.object,
  // Please check CardInfoFullColor props for more info
  rightCardInfoIconProps: _propTypes.default.object,
  // Please check CardInfoFullColor props for more info
  rightCardBlogImageProps: _propTypes.default.object,
  // Please check CardInfoFullColor props for more info
  rightCardContactUsTitleProps: _propTypes.default.object,
  // Please check CardInfoFullColor props for more info
  rightCardInfoImageProps: _propTypes.default.object
};
