// @notus-pro/react/ImagesOverlap.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ImagesOverlap;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ImageTooltip = _interopRequireDefault(require("./ImageTooltip.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function ImagesOverlap(_ref) {
  var images = _ref.images,
      text = _ref.text,
      size = _ref.size;
  var newSize = {
    sm: "w-6 h-6",
    lg: "w-10 h-10",
    regular: "w-8 h-8"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex items-center"
  }, /*#__PURE__*/_react.default.createElement("div", null, images.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("a", {
      className: "text-white bg-blueGray-500 inline-flex items-center justify-center shadow-lg rounded rounded-full relative border-2 border-white -ml-4 hover:z-1 " + newSize[size],
      key: key,
      href: "#pablo",
      onClick: function onClick(e) {
        return e.preventDefault();
      }
    }, /*#__PURE__*/_react.default.createElement(_ImageTooltip.default, prop));
  })), /*#__PURE__*/_react.default.createElement("small", {
    className: "pl-2 font-bold mb-1"
  }, text)));
}

ImagesOverlap.defaultProps = {
  images: [],
  size: "regular"
};
ImagesOverlap.propTypes = {
  images: _propTypes.default.arrayOf(_propTypes.default.object),
  text: _propTypes.default.string,
  size: _propTypes.default.oneOf(["sm", "lg", "regular"])
};
