// @notus-pro/react/InfoAreaUser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = InfoAreaUser;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable react/jsx-no-target-blank */
// components
function InfoAreaUser(_ref) {
  var image = _ref.image,
      name = _ref.name,
      jobTitle = _ref.jobTitle,
      socialLinks = _ref.socialLinks;
  var colors = {
    facebook: "bg-lightBlue-500",
    twitter: "bg-lightBlue-400",
    dribbble: "bg-pink-500",
    instagram: "bg-pink-700"
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "px-6"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "shadow-lg rounded-full mx-auto max-w-120-px"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "pt-6 text-center"
  }, /*#__PURE__*/_react.default.createElement("h5", {
    className: "text-xl font-bold"
  }, name), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-1 text-sm text-blueGray-400 uppercase font-bold"
  }, jobTitle), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-6"
  }, socialLinks.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("a", {
      key: key,
      href: prop.link,
      target: "_blank",
      type: "button",
      className: colors[prop.icon] + " inline-flex items-center justify-center text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1"
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: "fab fa-" + prop.icon
    }));
  }))));
}

InfoAreaUser.defaultProps = {
  socialLinks: []
};
InfoAreaUser.propTypes = {
  // src of the image
  image: _propTypes.default.string.isRequired,
  name: _propTypes.default.string,
  jobTitle: _propTypes.default.string,
  socialLinks: _propTypes.default.arrayOf( // this will generate an anchor with target blank to the given link
  _propTypes.default.shape({
    icon: _propTypes.default.oneOf(["facebook", "twitter", "instagram", "dribbble"]),
    link: _propTypes.default.string
  }))
};
