// @notus-pro/react/InfoAreaWhiteTitle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = InfoAreaWhiteTitle;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function InfoAreaWhiteTitle(_ref) {
  var icon = _ref.icon,
      type = _ref.type,
      title = _ref.title,
      description = _ref.description;
  var iconColors = "text-blueGray-900 bg-white";
  var titleColor = "text-white";
  var descriptionColor = "text-blueGray-500";

  if (type === "dark") {
    iconColors = "text-white bg-blueGray-900";
    titleColor = "text-blueGray-700";
    descriptionColor = "text-white opacity-75";
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: iconColors + " p-3 w-12 h-12 shadow-lg rounded-full inline-flex items-center justify-center"
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: icon + " text-xl"
  })), /*#__PURE__*/_react.default.createElement("h6", {
    className: titleColor + " text-xl mt-5 font-semibold"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: descriptionColor + " mt-2 mb-4"
  }, description));
}

InfoAreaWhiteTitle.defaultProps = {
  type: "light"
};
InfoAreaWhiteTitle.propTypes = {
  // icon name to be added on the <i></i> tag, for example "fa fa-heart"
  icon: _propTypes.default.string.isRequired,
  // choose the dark type when using this component on a lighter background
  // choose the light type when using this component on a darker background
  type: _propTypes.default.oneOf(["light", "dark"]),
  title: _propTypes.default.string,
  description: _propTypes.default.string
};
