// @notus-pro/react/MediaPlayerImageLeftRightButtons.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = MediaPlayerImageLeftRightButtons;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function MediaPlayerImageLeftRightButtons(_ref) {
  var items = _ref.items,
      defaultOpened = _ref.defaultOpened;

  var _React$useState = _react.default.useState(defaultOpened),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(false),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      oldInTransition = _React$useState4[0],
      setOldInTransition = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      newInTransition = _React$useState6[0],
      setNewInTransition = _React$useState6[1];

  var toggleNew = function toggleNew(e, newOpen) {
    e.preventDefault();

    if (!newInTransition && !oldInTransition) {
      setOldInTransition(true);
      setTimeout(function () {
        setOpen(newOpen);
      }, 500);
      setTimeout(function () {
        setOldInTransition(false);
        setNewInTransition(true);
      }, 600);
      setTimeout(function () {
        setNewInTransition(false);
      }, 1100);
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full overflow-hidden"
  }, items.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: (0, _classnames.default)("w-full p-12 transform duration-500 transition-all ease-in-out mx-auto", {
        hidden: key !== open,
        block: key === open,
        "opacity-0 scale-95": key === open && oldInTransition,
        "opacity-100 scale-100": key === open && newInTransition
      }),
      key: key
    }, /*#__PURE__*/_react.default.createElement("img", {
      alt: "...",
      src: prop,
      className: "w-full h-auto mx-auto"
    }));
  })), /*#__PURE__*/_react.default.createElement("a", {
    href: "#pablo",
    className: "absolute z-1 top-0 bottom-0 flex items-center justify-center w-1/6 text-white text-center opacity-75 hover:opacity-100 transition-opacity duration-150 ease-linear left-0",
    onClick: function onClick(e) {
      return toggleNew(e, open - 1 < 0 ? items.length - 1 : open - 1);
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-sm bg-orange-500 rounded-full fas fa-chevron-left flex items-center justify-center w-8 h-8"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sr-only"
  }, "Previous")), /*#__PURE__*/_react.default.createElement("a", {
    href: "#pablo",
    className: "absolute z-1 top-0 bottom-0 flex items-center justify-center w-1/6 text-white text-center opacity-75 hover:opacity-100 transition-opacity duration-150 ease-linear right-0",
    onClick: function onClick(e) {
      return toggleNew(e, open + 1 > items.length - 1 ? 0 : open + 1);
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-sm bg-orange-500 rounded-full fas fa-chevron-right flex items-center justify-center w-8 h-8"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sr-only"
  }, "Next"))));
}

MediaPlayerImageLeftRightButtons.defaultProps = {
  defaultOpened: 0,
  items: []
};
MediaPlayerImageLeftRightButtons.propTypes = {
  // 0 represents the first element
  // also, you should note that
  // the number should not be lower then 0
  // or higher than the number of items - 1
  defaultOpened: _propTypes.default.number,
  // an array of string representing valid image sources
  items: _propTypes.default.arrayOf(_propTypes.default.string)
};
