// @notus-pro/react/MediaPlayerTeamInfoSocial.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = MediaPlayerTeamInfoSocial;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function MediaPlayerTeamInfoSocial(_ref) {
  var items = _ref.items;

  var _React$useState = _react.default.useState(0),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(false),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      oldInTransition = _React$useState4[0],
      setOldInTransition = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      newInTransition = _React$useState6[0],
      setNewInTransition = _React$useState6[1];

  var toggleNew = function toggleNew(e, newOpen) {
    e.preventDefault();

    if (!newInTransition && !oldInTransition) {
      setOldInTransition(true);
      setTimeout(function () {
        setOpen(newOpen);
      }, 500);
      setTimeout(function () {
        setOldInTransition(false);
        setNewInTransition(true);
      }, 600);
      setTimeout(function () {
        setNewInTransition(false);
      }, 1100);
    }
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full overflow-hidden"
  }, items.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: (0, _classnames.default)("relative float-left w-full transform duration-500 transition-all ease-in-out", {
        hidden: key !== open,
        block: key === open,
        "opacity-0 scale-95": key === open && oldInTransition,
        "opacity-100 scale-100": key === open && newInTransition
      })
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "container mx-auto px-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex flex-wrap -mx-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "px-12 py-12 px-4 relative w-full md:w-5/12 ml-auto"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "p-6"
    }, /*#__PURE__*/_react.default.createElement("img", {
      className: "w-full rounded-lg shadow-xl transform",
      src: prop.image,
      alt: "..."
    }))), /*#__PURE__*/_react.default.createElement("div", {
      className: "pt-12 px-4 relative w-full md:w-5/12 mr-auto flex align-center items-center"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "p-4 pt-6 md:p-0"
    }, /*#__PURE__*/_react.default.createElement("h2", {
      className: "text-2xl font-bold mt-0 mb-2"
    }, prop.title), /*#__PURE__*/_react.default.createElement("p", {
      className: "mt-3 text-lg leading-relaxed text-blueGray-500"
    }, prop.description), /*#__PURE__*/_react.default.createElement("ul", {
      className: "mt-6 list-none"
    }, prop.list && prop.list.map(function (listProp, listKey) {
      return /*#__PURE__*/_react.default.createElement("li", {
        className: "py-2",
        key: listKey
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "flex items-center"
      }, /*#__PURE__*/_react.default.createElement("span", {
        className: "inline-flex text-center rounded-full mr-3 items-center justify-center text-sm w-8 h-8 text-" + listProp.color + "-500 bg-" + listProp.color + "-200"
      }, /*#__PURE__*/_react.default.createElement("i", {
        className: listProp.icon
      })), /*#__PURE__*/_react.default.createElement("h6", {
        className: "text-md text-blueGray-500"
      }, listProp.text)));
    })), /*#__PURE__*/_react.default.createElement("div", {
      className: "py-6 pl-12 text-left"
    }, prop.socials && prop.socials.map(function (socialsProp, socialsKey) {
      return /*#__PURE__*/_react.default.createElement(_Button.default, {
        key: socialsKey,
        href: socialsProp.link,
        color: socialsProp.icon
      }, /*#__PURE__*/_react.default.createElement("i", {
        className: "fab fa-" + socialsProp.icon
      }));
    })))))));
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full -mt-12"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#pablo",
    className: "absolute z-1 top-0 bottom-0 flex items-center justify-center w-1/6 text-white text-center opacity-50 hover:opacity-100 transition-opacity duration-150 ease-linear left-0",
    style: {
      left: "90%",
      width: "5%"
    },
    onClick: function onClick(e) {
      return toggleNew(e, open - 1 < 0 ? items.length - 1 : open - 1);
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-lightBlue-500 fas fa-chevron-left"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sr-only"
  }, "Previous")), /*#__PURE__*/_react.default.createElement("a", {
    href: "#pablo",
    className: "absolute z-1 top-0 bottom-0 flex items-center justify-center w-1/6 text-white text-center opacity-50 hover:opacity-100 transition-opacity duration-150 ease-linear right-0",
    style: {
      width: "5%"
    },
    onClick: function onClick(e) {
      return toggleNew(e, open + 1 > items.length - 1 ? 0 : open + 1);
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-lightBlue-500 fas fa-chevron-right"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sr-only"
  }, "Next"))));
}

MediaPlayerTeamInfoSocial.defaultProps = {
  items: []
};
MediaPlayerTeamInfoSocial.propTypes = {
  items: _propTypes.default.arrayOf(_propTypes.default.shape({
    title: _propTypes.default.string,
    description: _propTypes.default.string,
    image: _propTypes.default.string,
    socials: _propTypes.default.arrayOf( // this will generate an anchor with target blank to the given link
    _propTypes.default.shape({
      icon: _propTypes.default.oneOf(["facebook", "twitter", "instagram", "dribbble", "github"]),
      link: _propTypes.default.string
    })),
    list: _propTypes.default.arrayOf(_propTypes.default.shape({
      icon: _propTypes.default.string,
      text: _propTypes.default.string,
      color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
    }))
  }))
};
