// @notus-pro/react/MediaPlayerTeamInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = MediaPlayerTeamInfo;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function MediaPlayerTeamInfo(_ref) {
  var items = _ref.items,
      defaultOpened = _ref.defaultOpened;

  var _React$useState = _react.default.useState(defaultOpened),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(false),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      oldInTransition = _React$useState4[0],
      setOldInTransition = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      newInTransition = _React$useState6[0],
      setNewInTransition = _React$useState6[1];

  var toggleNew = function toggleNew(e, newOpen) {
    e.preventDefault();

    if (!newInTransition && !oldInTransition) {
      setOldInTransition(true);
      setTimeout(function () {
        setOpen(newOpen);
      }, 500);
      setTimeout(function () {
        setOldInTransition(false);
        setNewInTransition(true);
      }, 600);
      setTimeout(function () {
        setNewInTransition(false);
      }, 1100);
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-12 relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full overflow-hidden"
  }, /*#__PURE__*/_react.default.createElement("div", null, items.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: (0, _classnames.default)("p-6 transform duration-300 transition-all ease-in-out mx-auto", {
        hidden: key !== open,
        block: key === open,
        "opacity-0 scale-95": key === open && oldInTransition,
        "opacity-100 scale-100": key === open && newInTransition
      }),
      key: key
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "container mx-auto px-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "flex flex-wrap -mx-4"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "px-4 relative w-full md:w-6/12 flex items-center mx-auto"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "p-2"
    }, /*#__PURE__*/_react.default.createElement("h2", {
      className: "text-4xl font-bold leading-tight mt-0 mb-2"
    }, prop.title), /*#__PURE__*/_react.default.createElement("p", {
      className: "mt-4 mb-12 text-blueGray-500 text-lg leading-relaxed"
    }, prop.description), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, prop.button, {
      color: prop.color
    })))), /*#__PURE__*/_react.default.createElement("div", {
      className: "px-12 py-12 relative w-full md:w-5/12"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "p-6"
    }, /*#__PURE__*/_react.default.createElement("img", {
      className: "w-full rounded-lg shadow-xl",
      src: prop.image,
      alt: "..."
    }))))));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center mb-12"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#pablo",
    className: "text-white text-center opacity-50 hover:opacity-100 transition-opacity duration-150 ease-linear w-12 text-xl",
    onClick: function onClick(e) {
      return toggleNew(e, open - 1 < 0 ? items.length - 1 : open - 1);
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-lightBlue-500 fas fa-chevron-left"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sr-only"
  }, "Previous")), /*#__PURE__*/_react.default.createElement("a", {
    href: "#pablo",
    className: "text-white text-center opacity-50 hover:opacity-100 transition-opacity duration-150 ease-linear w-12 text-xl",
    onClick: function onClick(e) {
      return toggleNew(e, open + 1 > items.length - 1 ? 0 : open + 1);
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "text-lightBlue-500 fas fa-chevron-right"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "sr-only"
  }, "Next"))))));
}

MediaPlayerTeamInfo.defaultProps = {
  defaultOpened: 0,
  items: []
};
MediaPlayerTeamInfo.propTypes = {
  // 0 represents the first element
  // also, you should note that
  // the number should not be lower then 0
  // or higher than the number of items - 1
  defaultOpened: _propTypes.default.number,
  // an array of string representing valid image sources
  items: _propTypes.default.arrayOf(_propTypes.default.shape({
    image: _propTypes.default.string,
    title: _propTypes.default.string,
    description: _propTypes.default.string,
    // props to pass to the Button element
    // NOTE: the color is default set by the color prop
    button: _propTypes.default.object,
    color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
  }))
};
