// @notus-pro/react/Message.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Message;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Message(_ref) {
  var align = _ref.align,
      content = _ref.content,
      time = _ref.time,
      color = _ref.color,
      link = _ref.link;
  var bgClasses = {
    light: "bg-blueGray-100 text-blueGray-800",
    dark: "bg-blueGray-800 text-white",
    blueGray: "bg-blueGray-500 text-white",
    red: "bg-red-500 text-white",
    orange: "bg-orange-500 text-white",
    amber: "bg-amber-500 text-white",
    emerald: "bg-emerald-500 text-white",
    teal: "bg-teal-500 text-white",
    lightBlue: "bg-lightBlue-500 text-white",
    indigo: "bg-indigo-500 text-white",
    purple: "bg-purple-500 text-white",
    pink: "bg-pink-500 text-white"
  };
  var alignClasses = {
    left: "justify-start text-left",
    right: "justify-end text-right"
  };

  var details = /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-auto"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative flex px-4 py-2 my-4 break-words shadow-lg rounded-lg " + bgClasses[color]
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: ""
  }, content.map(function (prop, key) {
    return prop.image ? /*#__PURE__*/_react.default.createElement("img", {
      key: key,
      src: prop.image,
      className: "rounded my-2 max-w-250-px",
      alt: "..."
    }) : /*#__PURE__*/_react.default.createElement("p", {
      key: key,
      className: "mb-1 text-lg leading-relaxed"
    }, prop);
  }), /*#__PURE__*/_react.default.createElement("p", {
    className: "block uppercase font-semibold opacity-75 text-sm"
  }, time && align === "left" && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("i", {
    className: time.icon + " mr-2"
  }), /*#__PURE__*/_react.default.createElement("span", null, time.text)), time && align === "right" && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("span", {
    className: "mr-2"
  }, time.text), /*#__PURE__*/_react.default.createElement("i", {
    className: time.icon
  }))))));

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: alignClasses[align] + " flex flex-wrap -mx-4"
  }, link && link.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, link, details), link && link.to === undefined && /*#__PURE__*/_react.default.createElement("a", link, details)));
}

Message.defaultProps = {
  content: [],
  color: "blueGray",
  align: "right"
};
Message.propTypes = {
  align: _propTypes.default.oneOf(["left", "right"]),
  content: _propTypes.default.arrayOf(_propTypes.default.oneOfType([// this will generate a paragraph
  _propTypes.default.string, // this will generate an image
  _propTypes.default.shape({
    image: _propTypes.default.string
  })])),
  time: _propTypes.default.shape({
    icon: _propTypes.default.string,
    text: _propTypes.default.string
  }),
  color: _propTypes.default.oneOf(["white", "light", "dark", "blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  // properties to pass to the link object
  // that wrapps the whole component
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  link: _propTypes.default.object
};
