// @notus-pro/react/NavbarListLink.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = NavbarListLink;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function NavbarListLink(_ref) {
  var children = _ref.children,
      className = _ref.className,
      breakpoint = _ref.breakpoint,
      type = _ref.type,
      to = _ref.to,
      rest = _objectWithoutProperties(_ref, ["children", "className", "breakpoint", "type", "to"]);

  var types = {
    light: "text-white",
    dark: "text-blueGray-800"
  };
  var breakpoints = {
    all: "py-2",
    sm: "py-4 sm:py-2",
    md: "py-4 md:py-2",
    lg: "py-4 lg:py-2",
    xl: "py-4 xl:py-2"
  };
  var classes = (0, _classnames.default)("hover:opacity-75 px-3 flex items-center text-xs uppercase font-bold transition-all duration-150 ease-in-out", className, types[type], breakpoints[breakpoint]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, rest, {
    to: to,
    className: classes
  }), children), to === undefined && /*#__PURE__*/_react.default.createElement("a", _extends({}, rest, {
    className: classes
  }), children));
}

NavbarListLink.defaultProps = {
  className: "",
  type: "light"
};
NavbarListLink.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  // light is for white text
  // dark is for text-blueGray-800
  type: _propTypes.default.oneOf(["light", "dark"]),
  // where the padding should start being smaller
  breakpoint: _propTypes.default.oneOf(["all", "sm", "md", "lg", "xl"])
};
