// @notus-pro/react/NavbarSearchUser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = NavbarSearchUser;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _classnames = _interopRequireDefault(require("classnames"));

var _DropdownUser = _interopRequireDefault(require("./DropdownUser.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function NavbarSearchUser(_ref) {
  var brand = _ref.brand,
      input = _ref.input,
      dropdown = _ref.dropdown;

  var _React$useState = _react.default.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      collapseOpen = _React$useState2[0],
      setCollapseOpen = _React$useState2[1];

  var _React$useState3 = _react.default.useState(undefined),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      collapseStyle = _React$useState4[0],
      setCollapseStyle = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      animation = _React$useState6[0],
      setAnimation = _React$useState6[1];

  var collapseRef = _react.default.useRef(null);

  var startAnitmation = function startAnitmation() {
    if (!animation) {
      setAnimation(true);

      if (collapseOpen) {
        setCollapseStyle(0);
        setTimeout(function () {
          setAnimation(false);
          setCollapseOpen(false);
        }, 310);
      } else {
        setCollapseOpen(true);
        setTimeout(function () {
          setCollapseStyle(collapseRef.current.scrollHeight);
        }, 10);
        setTimeout(function () {
          setAnimation(false);
        }, 310);
      }
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("nav", {
    className: "absolute top-0 left-0 w-full z-10 bg-white lg:flex-row lg:flex-nowrap lg:justify-start flex items-center py-1 px-4 lg:bg-transparent"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "w-full mx-aut0 items-center flex justify-between lg:flex-nowrap flex-wrap lg:px-6 px-4"
  }, brand && brand.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, brand, {
    className: "text-blueGray-800 lg:text-white text-sm uppercase inline-block font-semibold my-3"
  }), brand.children), brand && brand.to === undefined && /*#__PURE__*/_react.default.createElement("a", _extends({}, brand, {
    className: "text-blueGray-800 lg:text-white text-sm uppercase inline-block font-semibold my-3"
  }), brand.children), /*#__PURE__*/_react.default.createElement("button", {
    className: "ml-auto cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-blueGray-400 rounded bg-transparent block outline-none focus:outline-none text-blueGray-300 lg:hidden",
    type: "button",
    onClick: startAnitmation
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "fas fa-bars"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)("items-center w-full lg:flex lg:w-auto flex-grow duration-300 transition-all ease-in-out lg:h-auto-important", {
      "block h-0": collapseOpen,
      hidden: !collapseOpen,
      "overflow-hidden": animation
    }),
    style: {
      height: collapseStyle
    },
    ref: collapseRef
  }, /*#__PURE__*/_react.default.createElement("form", {
    className: "flex flex-row flex-wrap items-center ml-auto mr-3 mt-3"
  }, /*#__PURE__*/_react.default.createElement(_Input.default, input)), /*#__PURE__*/_react.default.createElement(_DropdownUser.default, dropdown)))));
}

NavbarSearchUser.defaultProps = {
  brand: {},
  input: {},
  dropdown: {}
};
NavbarSearchUser.propTypes = {
  // properties to pass to the link object for the brand
  // if you pass a prop named to, the link will be
  // generated using Link from react-router-dom
  brand: _propTypes.default.object,
  // properties to pass to the Input component
  input: _propTypes.default.object,
  // properties to pass to the DropdownUser component
  dropdown: _propTypes.default.object
};
