// @notus-pro/react/PreFooterLarge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = PreFooterLarge;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function PreFooterLarge(_ref) {
  var title = _ref.title,
      subtitle = _ref.subtitle,
      socials = _ref.socials,
      type = _ref.type;
  var subtitleClass = "leading-normal mt-1 mb-2";
  var wrapperClass = "pt-12 pb-4 border-b";
  var wrapperTypes = {
    dark: "bg-blueGray-800 border-blueGray-700",
    light: "bg-blueGray-100"
  };
  var subtitleTypes = {
    dark: "text-white",
    light: "text-blueGray-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: wrapperClass + " " + wrapperTypes[type]
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "justify-between items-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-6/12"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    className: "text-xl font-bold leading-normal mt-0"
  }, title), /*#__PURE__*/_react.default.createElement("h4", {
    className: subtitleClass + " " + subtitleTypes[type]
  }, subtitle)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-right pr-6 px-4 relative w-full lg:w-6/12"
  }, socials.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, prop.buttonProps, {
      key: key
    }), /*#__PURE__*/_react.default.createElement("i", {
      className: prop.icon
    }));
  }))))));
}

PreFooterLarge.defaultProps = {
  socials: [],
  type: "light"
};
PreFooterLarge.propTypes = {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  type: _propTypes.default.oneOf(["dark", "light"]),
  socials: _propTypes.default.arrayOf(_propTypes.default.shape({
    icon: _propTypes.default.string,
    // the buttonProps can be any of the props of the Button component
    buttonProps: _propTypes.default.object
  }))
};
