// @notus-pro/react/Pricing1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Pricing1;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _CardPricing = _interopRequireDefault(require("./CardPricing.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function Pricing1(_ref) {
  var title = _ref.title,
      firstButtonText = _ref.firstButtonText,
      secondButtonText = _ref.secondButtonText,
      defaultOpen = _ref.defaultOpen,
      description = _ref.description,
      cardinfoiconProps = _ref.cardinfoiconProps,
      cardblogimageProps = _ref.cardblogimageProps,
      cardcontactustitleProps = _ref.cardcontactustitleProps,
      cardinfoimageProps = _ref.cardinfoimageProps;

  var _React$useState = _react.default.useState(1),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      openTab = _React$useState2[0],
      setOpenTab = _React$useState2[1];

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-12 bg-blueGray-100"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-center flex-col px-4 relative w-full md:w-4/12 ml-auto"
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: "text-3xl font-bold mt-0"
  }, title), /*#__PURE__*/_react.default.createElement("p", {
    className: "mt-2 mb-10 text-lg leading-relaxed text-blueGray-400"
  }, description), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_Button.default, {
    color: openTab === 1 ? "pink" : "white",
    onClick: function onClick() {
      return setOpenTab(1);
    }
  }, firstButtonText), /*#__PURE__*/_react.default.createElement(_Button.default, {
    color: openTab === 2 ? "pink" : "white",
    onClick: function onClick() {
      return setOpenTab(2);
    }
  }, secondButtonText))), /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto px-4 relative w-full md:w-7/12 w-full lg:w-7/12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "block"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: openTab === 1 ? "block" : "hidden"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardPricing.default, cardinfoiconProps)), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardPricing.default, cardblogimageProps)))), /*#__PURE__*/_react.default.createElement("div", {
    className: openTab === 2 ? "block" : "hidden"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardPricing.default, cardcontactustitleProps)), /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full md:w-6/12"
  }, /*#__PURE__*/_react.default.createElement(_CardPricing.default, cardinfoimageProps)))))))));
}

Pricing1.defaultProps = {
  defaultOpen: 1
};
Pricing1.propTypes = {
  title: _propTypes.default.string,
  firstButtonText: _propTypes.default.string,
  secondButtonText: _propTypes.default.string,
  defaultOpen: _propTypes.default.oneOf([1, 2]),
  description: _propTypes.default.string,
  // is represented by the props that you can pass
  // to our CardPricing components
  cardinfoiconProps: _propTypes.default.object,
  // is represented by the props that you can pass
  // to our CardPricing components
  cardblogimageProps: _propTypes.default.object,
  // is represented by the props that you can pass
  // to our CardPricing components
  cardcontactustitleProps: _propTypes.default.object,
  // is represented by the props that you can pass
  // to our CardPricing components
  cardinfoimageProps: _propTypes.default.object
};
