// @notus-pro/react/Pricing7.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Pricing7;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading3Badge = _interopRequireDefault(require("./Heading3Badge.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Pricing7(_ref) {
  var heading3badge = _ref.heading3badge,
      table = _ref.table;
  var colors = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading3badge && /*#__PURE__*/_react.default.createElement(_Heading3Badge.default, heading3badge), /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "block w-full overflow-x-auto"
  }, /*#__PURE__*/_react.default.createElement("table", {
    className: "w-full mb-4 border-collapse"
  }, /*#__PURE__*/_react.default.createElement("thead", {
    className: "text-blueGray-700"
  }, /*#__PURE__*/_react.default.createElement("tr", {
    className: "border-b"
  }, table.head.map(function (prop, key) {
    return typeof prop === "string" && /*#__PURE__*/_react.default.createElement("th", {
      key: key,
      className: (key === 0 ? "text-left" : "text-center") + " py-2 align-bottom p-3 border-0"
    }, /*#__PURE__*/_react.default.createElement("h5", {
      className: "text-2xl font-normal leading-normal mt-0 mb-2"
    }, prop)) || /*#__PURE__*/_react.default.createElement("th", {
      key: key,
      className: "text-center py-4 p-3 border-0"
    }, /*#__PURE__*/_react.default.createElement("p", {
      className: "text-blueGray-500 mb-0"
    }, prop.subtitle), /*#__PURE__*/_react.default.createElement("h4", {
      className: "text-3xl leading-tight mt-0 mb-6"
    }, prop.title), /*#__PURE__*/_react.default.createElement(_Button.default, prop.button));
  }))), /*#__PURE__*/_react.default.createElement("tbody", {
    className: "text-blueGray-500"
  }, table.body.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      key: key
    }, (prop || []).map(function (colProp, colKey) {
      return typeof colProp === "string" && /*#__PURE__*/_react.default.createElement("td", {
        key: colKey,
        className: (colKey !== 0 ? "text-center " : "px-3 text-left ") + "py-4"
      }, colProp) || /*#__PURE__*/_react.default.createElement("td", {
        key: colKey,
        className: "text-center py-4"
      }, /*#__PURE__*/_react.default.createElement("i", {
        className: colors[colProp.color] + " text-lg " + colProp.icon
      }));
    }));
  })))))))));
}

Pricing7.defaultProps = {
  table: {
    head: [],
    body: []
  },
  heading3badge: null
};
Pricing7.propTypes = {
  // prop to pass to the Heading3Badge component
  heading3badge: _propTypes.default.object,
  table: _propTypes.default.shape({
    head: _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.shape({
      title: _propTypes.default.string,
      subtitle: _propTypes.default.string,
      // props to pass to the Button component
      button: _propTypes.default.object
    })])),
    body: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.shape({
      icon: _propTypes.default.string,
      color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
    })])))
  })
};
