// @notus-pro/react/Product1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Product1;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading = _interopRequireDefault(require("./Heading2.js"));

var _MediaPlayerImageLeftRightButtons = _interopRequireDefault(require("./MediaPlayerImageLeftRightButtons.js"));

var _Incrementer = _interopRequireDefault(require("./Incrementer.js"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Input = _interopRequireDefault(require("./Input.js"));

var _Select = _interopRequireDefault(require("./Select.js"));

var _Ratings = _interopRequireDefault(require("./Ratings.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Product1(_ref) {
  var heading2 = _ref.heading2,
      name = _ref.name,
      price = _ref.price,
      description = _ref.description,
      mediaplayer = _ref.mediaplayer,
      incrementer = _ref.incrementer,
      addToCart = _ref.addToCart,
      inputs = _ref.inputs,
      ratings = _ref.ratings;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("section", {
    className: "relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading2 && /*#__PURE__*/_react.default.createElement(_Heading.default, heading2), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto px-4 relative w-full lg:w-6/12 w-full md:w-full"
  }, /*#__PURE__*/_react.default.createElement(_MediaPlayerImageLeftRightButtons.default, mediaplayer)), /*#__PURE__*/_react.default.createElement("div", {
    className: "mr-auto px-4 relative w-full lg:w-6/12 w-full md:w-full"
  }, /*#__PURE__*/_react.default.createElement("h2", {
    className: "text-3xl font-bold leading-tight mt-0 mb-0"
  }, name), /*#__PURE__*/_react.default.createElement("div", {
    className: "pt-2"
  }, /*#__PURE__*/_react.default.createElement(_Ratings.default, ratings)), /*#__PURE__*/_react.default.createElement("h2", {
    className: "text-3xl font-normal mt-2 mb-2"
  }, price), /*#__PURE__*/_react.default.createElement("p", {
    className: "text-blueGray-500"
  }, description), /*#__PURE__*/_react.default.createElement("div", {
    className: "mt-12 mb-6 flex flex-wrap -mx-4"
  }, inputs.map(function (prop, key) {
    if (prop.input) {
      return /*#__PURE__*/_react.default.createElement("div", {
        key: key,
        className: "px-4 relative w-full lg:w-6/12"
      }, /*#__PURE__*/_react.default.createElement("label", {
        className: "inline-block mb-2"
      }, prop.label), /*#__PURE__*/_react.default.createElement(_Input.default, prop.input));
    } else if (prop.select) {
      return /*#__PURE__*/_react.default.createElement("div", {
        key: key,
        className: "px-4 relative w-full lg:w-6/12"
      }, /*#__PURE__*/_react.default.createElement("label", {
        className: "inline-block mb-2"
      }, prop.label), /*#__PURE__*/_react.default.createElement(_Select.default, prop.select));
    }

    return null;
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-6 flex flex-wrap -mx-4"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "px-4 relative w-full lg:w-5/12"
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "inline-block mb-2"
  }, "Quantity"), /*#__PURE__*/_react.default.createElement(_Incrementer.default, incrementer))), /*#__PURE__*/_react.default.createElement(_Button.default, addToCart))))));
}

Product1.defaultProps = {
  mediaplayer: {},
  incrementer: {},
  addToCart: {},
  inputs: [],
  ratings: {},
  heading2: null
};
Product1.propTypes = {
  // prop to pass to the Heading2 component
  heading2: _propTypes.default.object,
  name: _propTypes.default.string,
  price: _propTypes.default.string,
  description: _propTypes.default.string,
  // props to pass to the MediaPlayerImageLeftRightButtons component
  mediaplayer: _propTypes.default.object,
  // props to pass to the Incrementer component
  incrementer: _propTypes.default.object,
  // props to pass to the Button component
  addToCart: _propTypes.default.object,
  // props to pass to the Ratings component
  ratings: _propTypes.default.object,
  inputs: _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.shape({
    label: _propTypes.default.string,
    // props to pass to the Input component
    input: _propTypes.default.object
  }), _propTypes.default.shape({
    label: _propTypes.default.string,
    // props to pass to the Select component
    select: _propTypes.default.object
  })]))
};
