// @notus-pro/react/Progress.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Progress;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Badge = _interopRequireDefault(require("./Badge.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Progress(_ref) {
  var value = _ref.value,
      text = _ref.text,
      color = _ref.color;
  var progress = {
    blueGray: "bg-blueGray-500",
    red: "bg-red-500",
    orange: "bg-orange-500",
    amber: "bg-amber-500",
    emerald: "bg-emerald-500",
    teal: "bg-teal-500",
    lightBlue: "bg-lightBlue-500",
    indigo: "bg-indigo-500",
    purple: "bg-purple-500",
    pink: "bg-pink-500"
  };
  var percent = {
    blueGray: "text-blueGray-500",
    red: "text-red-500",
    orange: "text-orange-500",
    amber: "text-amber-500",
    emerald: "text-emerald-500",
    teal: "text-teal-500",
    lightBlue: "text-lightBlue-500",
    indigo: "text-indigo-500",
    purple: "text-purple-500",
    pink: "text-pink-500"
  };
  var wrapper = {
    blueGray: "bg-blueGray-200",
    red: "bg-red-200",
    orange: "bg-orange-200",
    amber: "bg-amber-200",
    emerald: "bg-emerald-200",
    teal: "bg-teal-200",
    lightBlue: "bg-lightBlue-200",
    indigo: "bg-indigo-200",
    purple: "bg-purple-200",
    pink: "bg-pink-200"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative w-full"
  }, text && /*#__PURE__*/_react.default.createElement("div", {
    className: "flex mb-2 items-center justify-between"
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_Badge.default, {
    color: color,
    round: true
  }, text)), /*#__PURE__*/_react.default.createElement("div", {
    className: "text-right"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "text-xs font-semibold inline-block " + percent[color]
  }, value, "%"))), /*#__PURE__*/_react.default.createElement("div", {
    className: "overflow-hidden h-2 text-xs flex rounded " + wrapper[color]
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: value.toString() + "%"
    },
    className: "shadow-none flex flex-col text-center whitespace-nowrap text-white justify-center " + progress[color]
  }))));
}

Progress.defaultProps = {
  color: "blueGray",
  value: 0
};
Progress.propTypes = {
  text: _propTypes.default.string,
  value: _propTypes.default.number,
  color: _propTypes.default.oneOf(["blueGray", "red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"])
};
