// @notus-pro/react/Projects2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Projects2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading = _interopRequireDefault(require("./Heading2.js"));

var _CardProjectTeam = _interopRequireDefault(require("./CardProjectTeam.js"));

var _Pagination = _interopRequireDefault(require("./Pagination.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Projects2(_ref) {
  var heading2 = _ref.heading2,
      heading2badge = _ref.heading2badge,
      heading3 = _ref.heading3,
      heading3badge = _ref.heading3badge,
      cards = _ref.cards,
      pagination = _ref.pagination,
      footerText = _ref.footerText;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "py-12"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading2 && /*#__PURE__*/_react.default.createElement(_Heading.default, heading2), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12 flex flex-wrap -mx-4"
  }, cards.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "px-4 relative w-full md:w-4/12"
    }, /*#__PURE__*/_react.default.createElement(_CardProjectTeam.default, prop));
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex justify-between items-center"
  }, /*#__PURE__*/_react.default.createElement(_Pagination.default, pagination), /*#__PURE__*/_react.default.createElement("p", null, footerText)))));
}

Projects2.defaultProps = {
  cards: [],
  pagination: {},
  heading2: null
};
Projects2.propTypes = {
  // prop to pass to the Heading2 component
  heading2: _propTypes.default.object,
  // array of props to pass to the CardProjectTeam component
  cards: _propTypes.default.arrayOf(_propTypes.default.object),
  // props to pass to the Pagination component
  pagination: _propTypes.default.object,
  footerText: _propTypes.default.string
};
