// @notus-pro/react/Select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Select;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@popperjs/core");

var _Input = _interopRequireDefault(require("./Input.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Select(_ref) {
  var border = _ref.border,
      size = _ref.size,
      leftIcon = _ref.leftIcon,
      rightIcon = _ref.rightIcon,
      type = _ref.type,
      items = _ref.items,
      closeOnSelect = _ref.closeOnSelect,
      placeholder = _ref.placeholder,
      defaultValue = _ref.defaultValue,
      rest = _objectWithoutProperties(_ref, ["border", "size", "leftIcon", "rightIcon", "type", "items", "closeOnSelect", "placeholder", "defaultValue"]);

  var _React$useState = _react.default.useState(defaultValue),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      inputValue = _React$useState2[0],
      setInputValue = _React$useState2[1];

  var _React$useState3 = _react.default.useState(""),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      menuClasses = _React$useState4[0],
      setMenuClasses = _React$useState4[1];

  var _React$useState5 = _react.default.useState(false),
      _React$useState6 = _slicedToArray(_React$useState5, 2),
      menuShow = _React$useState6[0],
      setMenuShow = _React$useState6[1];

  var _React$useState7 = _react.default.useState(false),
      _React$useState8 = _slicedToArray(_React$useState7, 2),
      animating = _React$useState8[0],
      setAntimating = _React$useState8[1];

  var _React$useState9 = _react.default.useState("origin-top-right"),
      _React$useState10 = _slicedToArray(_React$useState9, 2),
      transformOrigin = _React$useState10[0],
      setTransformOrigin = _React$useState10[1];

  var btnDropdownRef = /*#__PURE__*/_react.default.createRef();

  var popoverDropdownRef = /*#__PURE__*/_react.default.createRef();

  var origins = {
    "top-start": "origin-bottom-left",
    "top-end": "origin-bottom-right",
    "bottom-start": "origin-top-left",
    "bottom-end": "origin-top-right",
    "right-start": "origin-top-left",
    "right-end": "origin-top-left",
    "left-start": "origin-top-right",
    "left-end": "origin-top-right",
    bottom: "origin-top",
    top: "origin-bottom",
    left: "origin-right",
    right: "origin-left"
  };

  var startAnimation = function startAnimation() {
    if (!animating) {
      setAntimating(true);

      if (menuShow) {
        setMenuClasses("");
        setTimeout(function () {
          setMenuShow(false);
        }, 310);
      } else {
        var popperObject = (0, _core.createPopper)(btnDropdownRef.current, popoverDropdownRef.current, {
          placement: "bottom-start"
        });
        setTransformOrigin(origins[popperObject.state.placement]);
        setMenuShow(true);
        setTimeout(function () {
          setMenuClasses("opacity-100 scale-100 ");
        }, 10);
      }

      setTimeout(function () {
        setAntimating(false);
      }, 350);
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "relative"
  }, /*#__PURE__*/_react.default.createElement(_Input.default, {
    value: inputValue,
    type: "text",
    onChange: function onChange() {},
    onClick: startAnimation,
    placeholder: placeholder,
    ref: btnDropdownRef,
    size: size
  }), /*#__PURE__*/_react.default.createElement("div", {
    ref: popoverDropdownRef,
    className: (menuShow ? "block " : "hidden ") + "z-50"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: menuClasses + " absolute opacity-0 border bg-white text-base z-50 float-left p-2 text-left rounded-lg shadow-lg min-w-48 transition-all duration-200 ease-in-out transform scale-95 origin-top-left " + transformOrigin
  }, items.map(function (prop, key) {
    if (prop.disabled) {
      return /*#__PURE__*/_react.default.createElement("span", {
        className: "text-sm pt-2 pb-0 px-4 font-bold block w-full whitespace-nowrap bg-transparent text-blueGray-500",
        key: key
      }, prop.text);
    } else {
      return /*#__PURE__*/_react.default.createElement("a", {
        href: "#pablo",
        key: key,
        className: (prop === inputValue ? "bg-lightBlue-100 " : "") + "text-sm px-3 py-2 font-normal block w-full whitespace-nowrap transition-all duration-200 hover:bg-blueGray-100 rounded",
        onClick: function onClick(e) {
          e.preventDefault();
          setInputValue(prop);

          if (closeOnSelect) {
            startAnimation();
          }
        }
      }, prop);
    }
  })))));
}

Select.defaultProps = {
  border: "border",
  size: "regular",
  items: [],
  closeOnSelect: true,
  defaultValue: ""
};
Select.propTypes = {
  placeholder: _propTypes.default.string,
  defaultValue: _propTypes.default.string,
  border: _propTypes.default.oneOf(["border", "borderless"]),
  closeOnSelect: _propTypes.default.bool,
  items: _propTypes.default.arrayOf(_propTypes.default.oneOfType([// pass an object with text of string
  // if you want it to be disabled
  _propTypes.default.shape({
    text: _propTypes.default.string
  }), // this will not be disabled and
  // users will be able to select it
  _propTypes.default.string])),
  size: _propTypes.default.oneOf(["sm", "lg", "regular"]),
  // NOTE: you sould only pass icon classes
  // // // if you also pass tailwindcss classes
  // // // the output may not be a desired one
  leftIcon: _propTypes.default.string,
  // NOTE: you sould only pass icon classes
  // // // if you also pass tailwindcss classes
  // // // the output may not be a desired one
  rightIcon: _propTypes.default.string
};
