// @notus-pro/react/Sidebar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Sidebar;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames3 = _interopRequireDefault(require("classnames"));

var _reactRouterDom = require("react-router-dom");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function Sidebar(_ref) {
  var brand = _ref.brand,
      items = _ref.items,
      activeColor = _ref.activeColor;

  var _React$useState = _react.default.useState("-translate-x-full"),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      sidebarShow = _React$useState2[0],
      setSidebarShow = _React$useState2[1];

  var activeColors = {
    red: "text-red-500 hover:text-red-700",
    orange: "text-orange-500 hover:text-orange-700",
    amber: "text-amber-500 hover:text-amber-700",
    emerald: "text-emerald-500 hover:text-emerald-700",
    teal: "text-teal-500 hover:text-teal-700",
    lightBlue: "text-lightBlue-500 hover:text-lightBlue-700",
    indigo: "text-indigo-500 hover:text-indigo-700",
    purple: "text-purple-500 hover:text-purple-700",
    pink: "text-pink-500 hover:text-pink-700"
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("nav", {
    className: "block py-4 px-6 top-0 bottom-0 w-64 bg-white shadow-xl left-0 fixed flex-row flex-nowrap md:z-10 z-9999 transition-all duration-300 ease-in-out transform md:translate-x-0 " + sidebarShow
  }, /*#__PURE__*/_react.default.createElement("button", {
    className: "md:hidden flex items-center justify-center cursor-pointer text-blueGray-700 w-6 h-10 border-l-0 border-r border-t border-b border-solid border-blueGray-100 text-xl leading-none bg-white rounded-r border border-solid border-transparent absolute top-1/2 -right-24-px focus:outline-none z-9998",
    onClick: function onClick() {
      if (sidebarShow === "") {
        setSidebarShow("-translate-x-full");
      } else {
        setSidebarShow("");
      }
    }
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "fas fa-ellipsis-v"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-col min-h-full px-0 flex flex-wrap items-center justify-between w-full mx-auto overflow-y-auto overflow-x-hidden"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex bg-white flex-col items-stretch opacity-100 relative mt-4 overflow-y-auto overflow-x-hidden h-auto z-40 items-center flex-1 rounded w-full"
  }, brand && brand.link && brand.link.to && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, brand, {
    className: "md:flex items-center flex-col text-center md:pb-2 text-blueGray-700 mr-0 inline-flex whitespace-nowrap text-sm uppercase font-bold p-4 px-0"
  }), brand && brand.image && /*#__PURE__*/_react.default.createElement("img", {
    src: brand.image,
    alt: "...",
    className: "max-w-full rounded"
  }), brand && brand.text && /*#__PURE__*/_react.default.createElement("span", null, brand.text)), brand && brand.link && brand.link.to === undefined && /*#__PURE__*/_react.default.createElement("a", _extends({}, brand, {
    className: "md:flex items-center flex-col text-center md:pb-2 text-blueGray-700 mr-0 inline-flex whitespace-nowrap text-sm uppercase font-bold p-4 px-0"
  }), brand && brand.image && /*#__PURE__*/_react.default.createElement("img", {
    src: brand.image,
    alt: "...",
    className: "max-w-full rounded"
  }), brand && brand.text && /*#__PURE__*/_react.default.createElement("span", null, brand.text)), /*#__PURE__*/_react.default.createElement("div", {
    className: "md:flex-col md:min-w-full flex flex-col list-none"
  }, items.map(function (prop, key) {
    if (prop.divider) {
      return /*#__PURE__*/_react.default.createElement("hr", {
        key: key,
        className: "my-4 md:min-w-full"
      });
    } else if (prop.title) {
      return /*#__PURE__*/_react.default.createElement("h6", {
        key: key,
        className: "md:min-w-full text-blueGray-500 text-xs uppercase font-bold block pt-1 pb-4 no-underline"
      }, prop.title);
    } else if (prop.link && prop.link.to) {
      var _classnames;

      return /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, _extends({}, prop.link, {
        key: key,
        className: (0, _classnames3.default)("text-xs uppercase py-3 font-bold block", (_classnames = {}, _defineProperty(_classnames, activeColors[activeColor], prop.active), _defineProperty(_classnames, "text-blueGray-800 hover:text-blueGray-500", !prop.active || prop.active === undefined), _classnames))
      }), /*#__PURE__*/_react.default.createElement("i", {
        className: (0, _classnames3.default)(prop.icon, "mr-2 text-sm", {
          "opacity-75": prop.active,
          "text-blueGray-400": !prop.active || prop.active === undefined
        })
      }), prop.text);
    } else {
      var _classnames2;

      return /*#__PURE__*/_react.default.createElement("a", _extends({}, prop.link, {
        key: key,
        className: (0, _classnames3.default)("text-xs uppercase py-3 font-bold block", (_classnames2 = {}, _defineProperty(_classnames2, activeColors[activeColor], prop.active), _defineProperty(_classnames2, "text-blueGray-800 hover:text-blueGray-500", !prop.active || prop.active === undefined), _classnames2))
      }), /*#__PURE__*/_react.default.createElement("i", {
        className: (0, _classnames3.default)(prop.icon, "mr-2 text-sm", {
          "opacity-75": prop.active,
          "text-blueGray-400": !prop.active || prop.active === undefined
        })
      }), prop.text);
    }
  }))))));
}

Sidebar.defaultProps = {
  items: [],
  activeColor: "pink"
};
Sidebar.propTypes = {
  // this only applies for those items that have active set to true
  activeColor: _propTypes.default.oneOf(["red", "orange", "amber", "emerald", "teal", "lightBlue", "indigo", "purple", "pink"]),
  brand: _propTypes.default.shape({
    text: _propTypes.default.string,
    image: _propTypes.default.string,
    // props to pass to the wrapper link of the text and image
    // if you pass a prop named to, it will be generated as
    // Link from react-router-dom
    link: _propTypes.default.object
  }),
  items: _propTypes.default.arrayOf(_propTypes.default.oneOfType([// this will generate a divider hr tag
  _propTypes.default.shape({
    divider: _propTypes.default.bool
  }), // this will generate a text with the title string
  _propTypes.default.shape({
    title: _propTypes.default.string
  }), // this will generate a Link/Anchor with bellow options
  _propTypes.default.shape({
    icon: _propTypes.default.string,
    text: _propTypes.default.string,
    // if set to true, the link will change color to the active one
    active: _propTypes.default.bool,
    // props to pass to the wrapper link of the text and icon
    // if you pass a prop named to, it will be generated as
    // Link from react-router-dom
    link: _propTypes.default.object
  })]))
};
