// @notus-pro/react/Teams6.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Teams6;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading = _interopRequireDefault(require("./Heading3.js"));

var _CardProfileActions = _interopRequireDefault(require("./CardProfileActions.js"));

var _CardProfileReviews = _interopRequireDefault(require("./CardProfileReviews.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Teams6(_ref) {
  var heading3 = _ref.heading3,
      cards = _ref.cards;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "my-20"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading3 && /*#__PURE__*/_react.default.createElement(_Heading.default, heading3), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-wrap -mx-4"
  }, cards.map(function (prop, key) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: key,
      className: "px-4 relative w-full md:w-6/12 lg:w-4/12 my-4" + (key % 3 === 1 ? " mt-12" : "")
    }, prop && prop.cardprofilereviews && /*#__PURE__*/_react.default.createElement(_CardProfileReviews.default, _extends({
      key: key
    }, prop.cardprofilereviews)), prop && prop.cardprofileactions && /*#__PURE__*/_react.default.createElement(_CardProfileActions.default, _extends({
      key: key
    }, prop.cardprofileactions)));
  })))));
}

Teams6.defaultProps = {
  cards: [],
  heading3: null
};
Teams6.propTypes = {
  // prop to pass to the Heading3 component
  heading3: _propTypes.default.object,
  cards: _propTypes.default.arrayOf(_propTypes.default.oneOfType([// props that can be passed to the
  // the CardProfileActions component
  _propTypes.default.shape({
    cardprofileactions: _propTypes.default.object
  }), // props that can be passed to the
  // CardProfileReviews component
  _propTypes.default.shape({
    cardprofilereviews: _propTypes.default.object
  })]))
};
