// @notus-pro/react/Testimonials2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Testimonials2;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Heading3Badge = _interopRequireDefault(require("./Heading3Badge.js"));

var _MediaPlayerTeamInfo = _interopRequireDefault(require("./MediaPlayerTeamInfo.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function Testimonials2(_ref) {
  var heading3badge = _ref.heading3badge,
      skew = _ref.skew,
      mediaplayer = _ref.mediaplayer;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "pb-12 mt-12 bg-blueGray-100 relative"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "container mx-auto px-4"
  }, heading3badge && /*#__PURE__*/_react.default.createElement(_Heading3Badge.default, heading3badge), /*#__PURE__*/_react.default.createElement(_MediaPlayerTeamInfo.default, mediaplayer))));
}

Testimonials2.defaultProps = {
  button: {},
  skew: {},
  heading3badge: null
};
Testimonials2.propTypes = {
  // prop to pass to the Heading3Badge component
  heading3badge: _propTypes.default.object,
  // props to pass to the Skew component
  skew: _propTypes.default.object,
  // props to pass to the MediaPlayerTeamInfo component
  mediaplayer: _propTypes.default.object
};
