// @notus-pro/react/WriteComment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = WriteComment;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Input = _interopRequireDefault(require("./Input.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
function WriteComment(_ref) {
  var image = _ref.image,
      input = _ref.input;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "items-center flex items-start"
  }, /*#__PURE__*/_react.default.createElement("img", {
    alt: "...",
    src: image,
    className: "mr-4 rounded-full w-10 h-10"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-1 mt-6"
  }, /*#__PURE__*/_react.default.createElement("form", null, /*#__PURE__*/_react.default.createElement(_Input.default, input)))));
}

WriteComment.defaultProps = {
  input: {}
};
WriteComment.propTypes = {
  image: _propTypes.default.string,
  // It is represetnted by the props you
  // can pass to our Input component element
  // NOTE: if you wish to generate a textarea, you will need to pass
  // // // inside this object >> type: "textarea"
  // // // full example >> input: { type: "textarea" }
  input: _propTypes.default.object
};
